\graphicspath{{chapters/5.Chapter_3/figures}}

\chapter{Transporters}

\section{Introduction}

\section{Aims}

\section{Methods}

\subsection{Database download}

Kodama Trinity Assembly 


\subsection{Transporter Identification}

\subsection{qPCR}


\section{Results}
\subsection{Kodama Assembly}
\subsubsection{Assembly}



\subsection{Transporter identification}

To identify transporters present in the translated protein dataset of the 
endosymbiont bins of the CCAP1660/12 and YADGN1 
\textit{Paramecium bursaria} transcriptome assemblies, as well as the Chlorella NC64A 
and Coccoymxa C-169 predicted proteomes the following process was used:

\begin{enumerate}
        \item TMHMM was used to predict TM domains within each protein
        \item Proteins were then filtered on those of which had 3 or more TM domains
        \item Those proteins were then searched using HMM and the PFAM HMM database
        \item Those hits were parsed for those which shared a PFAM domain with TCDB proteins
\end{enumerate}

\section{Discussion}

\section{Conclusion}


Network analysis
http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3299011/pdf/fmicb-03-00085.pdf



%Genome assembly - spades macmanes trimming was worse than spades harash trimming (shorter contigs)
