\graphicspath{{chapters/5.Chapter_3/figures}}

\begin{savequote}[75mm]
``All models are wrong, but some are useful''
\qauthor{- George E.P. Box & Draper: \textit{Empirical model-building and response surfaces, 1987}}
\end{savequote}

\chapter{Metabolic integration}

\section{Introduction}

The linking of metabolism between host and endosymbiont is a fundamental 
stage in endosymbiotic integration \citep{Bhattacharya2007,Karkar2015a}.
This metabolic integration provides the fundamental selective 
benefits of such an association by allowing the complementation 
of respective metabolic deficiencies or limitations and thus
allow both host and endosymbiont to exploit novel 
niches \citep{Hoffmeister2003}.

The most important group of proteins in determining levels of metabolic
co-dependence are host and endosymbiont transporter proteins integrated
into the outermembrane of the endosymbiont and the inner endosymbiotic
membrane of the host i.e. the perialgal vacuole in the case of
\textit{P. bursaria} and its green algal endosymbionts. 


These transporters control the movement of metabolites between host
and endosymbiont.

Indeed, the evolution of protein import systems 
from host to endosymbiont, such as the TICs/TOCs and TIMs/TOMs
of plastids and mitochondria, is considered by many 
as the key constraint in the establishment of an endosymbiont
as an organelle \citep{Pfanner2001.Keeling2008a} 



These protein import systems are more complicated
in the cases of secondary and tertiary endosymbioses
than basic plsatids due to the increased number
of membranes that may need to be traversed especially
for import directly to the secondary plastid from the
host \citep{Hirakawa2012}




These systems form such a crucial component of the
organellar milieu as they allow the increased transfer
of endosymbiotic genes 



 The most obvious point of metabolic integration in any photosynthetic
 endosymbiosis is the exchange
 of photosynthate from the endosymbiont to the host.



The transfer of matlose, glucose, fructose and malate from endosymbiont
to host has been observed using radiolabelling 
\citep{Brown1974}.  Furthermore, green algae strains competent to
 form endosymbioses were found to inducibly release significantly
 more photosynthate (in the form of \(\sim 95\%\) maltose) than strictly free-living strains
 in the presence of \(NaHCO_3\) on the order of \(5.4-86.7\%\) vs. \(0.4-7.6\%\)
 of total photosynthate \citep{Muscatine1967a}.

 pH-dependent release of photosynthate \citep{}



 Respiration of the host plays a role in the gas exchange of \(CO_2\) and \(O_2\) 
 to the endosymbiont, with the algae displaying higher levels of photosynthetic
 activity while in association with the host due to the increase host related 
 \(CO_2\) respiration \citep{Reisser1980}.

Addition of glucose, which likely increases host respiration rate and thus \(CO_2\)
evolution increases the rate of photosynthetic oxygen production commensurately \citep{Reisser1980}






Nitrogen is the ost transferred material after carbon \citep{Kato2009}, 
A huge range of states and parametetrs have been studied showing a range
of nitrate annd potentuially maino acid based systems (particularl L-glutamine and ammonium \citep{Albers1982}).


Neither green or algae free \textit{P. bursaria} were found to take up nitrate 
Ammonia is excreted by host without algae \citep{Albers1982}.


Proton gradient dependent transport of maltose/photosynthate out of the \citep{Schussler1992}
pH induces release of maltose in many algae \citep{}




The exact nature varies 




These early results are unclear on specific strains and how they correspond
to modern taxa \citep{Kato2009}. 









%Transporters are necessary for the transport and localisation of effectors and metabolites
%that are too large to diffuse through the membrane and for active transportation
%against concentration gradients. 
%
%
%When considered from a purely quantitative and mass perspective, sugars
%are the most important organic compounds to be taken up by living organisms. 
%As well as their obvious role in energy processes 
%
%
%
%There are 4 archetypical 
%
%
%\subsection{Sugar Transporters}
%
%In \textit{Parachlorella kessleri} (previous \textit{Chlorella kessleri} \citep{Krienitz2004,Hoshina2010})
%
%Chlorella \(H^{+}\)-monosaccharide (HUP1) co-transporters \citep{Sauer1989}
%have been identified (via hexose-transport deficient mutant screens \citep{Sauer1986}
%and localised to the plasma membrane \citep{Stadler1995}
%
%
%Additionally these transporters have been identified in the
%\textit{Auxenochlorella protothecoides} (formerly \textit{Chlorella protothecoides} \citep{Champenois2014}) genome \citep{Gao2014}.
%
%
%
%
%\subsection{Amino acid transporters}



There are 3 previously established points of metabolic contact between 
\textit{P. bursaria} and its green algal endosymbionts. 

Namely, sugar transport (mainly maltose), 

By analysis of a F36-ZK \textit{C. variabilis} \citep{Hoshina2010} endosymbiont
that 


Other \textit{C. variabilis} strais 
and \textit{M. reisseri} have previously been found to utilise nitrate 
in the same manner has other plants and algae as a nitrogen
source by the conversion to ammonium by nitrate reductase (NR)
and nitrite reductase (NiR).


What do \textit{Coccomyxa} sp. and \textit{C. vulgaris} use?



Japanese \textit{C. variabilis} 



\section{Aims}

The principal aim of this chapter is to identify
transporter proteins present in the endosymbiont
binned transcripts from the CCAP1660/12 RNA-Seq analysis,
a re-assembly of the \citep{Kodama2014c} dataset.

\section{Methods}

\subsection{Data Acquisition}

Sets of algal predicted proteins were acquired as follows:

\subsubsection{\textit{Chlorella variabilis} NC64A and \textit{Coccomyxa subellipsoidea} C-169} 
 
\textit{Coccomyxa subellipsoidea} C-169 genome project \citep{Blanc2012} version 2.0 
JGI annotated proteins (created 12-01-2014) were downloaded from JGI's
Phyotozome v10.3.1 \citep{Goodstein2012}. 
Similarly, the ``best'' annotated proteins from
version 1 of the \textit{Chlorella variabilis} NC64A genome project \citep{Blanc2010}
were downloaded from JGI's genome portal \citep{Grigoriev2011,Nordberg2014}

\subsubsection{\textit{Micractinium reisseri}}

Sequences were derived from endosymbiont binned
transcripts and annotations as described in \ref{chap:transcriptomics}.

\subsubsection{\textit{Chlorella variabilis 1 N}}

\(232.3M\) 100bp paired-end reads from \citep{Kodama2014}'s 
bulk RNAseq transcriptome of \textit{Paramecium bursaria} Yad1g (syngen
3, mating type 1) bearing \textit{Chlorella variabilis} 1N endosymbionts
were downloaded from the DNA Data Bank of Japan (DDBJ) \citep{Tateno2002,Kaminuma2011}
in Sequence Read Archive (SRA) format \citep{Leinonen2011,KodamaNRA2012b} (accession DRA000907 \citep{Kodama2014}).

These reads were then converted to fastq using ``fastq-dump'' using the SRA Toolkit
\citep{NationalCenterforBiotechnologyInformation2011}.  Reads were then trimmed
for sequencing adapters using ILLUMINACLIP and SLIDINGWINDOW with a window size
of 4 and a minimum average quality of 5 in Trimmomatic \citep{Bolger2014a}.

Reads were then error-corrected using ``SEECER'' with a k-mer size of 25 and 
default settings otherwise (entropy of 0.6 and a cluster log-likelihood
of -1) \citep{Le2013}.  Error-corrected reads were digitally normalised
using a K-mer size of 25 and a coverage of 20 \citep{Brown2012} and 
low abundance K-mers in high coverage reads were filtered \citep{Zhang2014,Zhang2015}
using the Khmer software package \citep{Doring2008,Crusoe2015}.

Assemblies were completed in a modified/fixed version of 
Bridger 2014-12-01 \citep{Chang2015} (available at
\url{https://github.com/fmaguire/Bridger_Assembler}) and 
Trinity v2.0.6 \citep{Grabherr2011,Haas2013} both with K-mer
sizes of 25.

An alternative Trinity assembly was also completed using
SLIDINGWINDOW Q30 trimmed reads without normalisation or 
error correction.

Assemblies were then compared using RSEM-EVAL \citep{Li2014} and the best
overall assembly selected on the basis of likelihood.

ORFs were called from the best assembly using universal and tetrahymena encodings 
via TransDecoder \citep{Haas2013} retaining the best scoring sequences and those
with HMMR hits to PFAM and BLASTP hits to the swissprot database. 

Phylogenies were generated for each sequence using the same approach and pipeline
described in Chapter 2. These phylogenies were subsequently classified using the 
same trained K-Neighbours supervised learning algorithm.
Any sequence that didn't have enough BLAST hits in the genomes used to generate
a phylogeny (5) were parsed based on what hits were retrieved.
Those with no hits were classified as ``unknown'' and those with
hits were classified based on the origin of those hits e.g. hits
to green algae and plant genomes were considered ``endosymbiont'' and so on.

Finally, the ORF bins for host and endosymbiont 
from both encodings were manually combined and reconciled
to generate transcript bins.  With the transcripts binned into
``host'' and ``endosymbiont'' ORFs were recalled from them using the appropriate
encodings. 

\subsection{Transporter identification}

Transporters were identified in the 4 sets of sequences (\textit{C. variabilis}, \textit{M. reisseri},
\textit{C. vulgaris} and \textit{C. subellipsoidea}) using the following set of pipelined filters:
\begin{enumerate}
    \item Transmembrane (TM) domains were predicted for each sequence using an HMM approach implemented as part of TMHMM2 \citep{Sonnhammer1998,Krogh2001}
and sequence predicted to contain at least 1 TM domain was extracted.
    \item These sequences were then used to search a PFAM database of profile HMMs \citep{Eddy1998} via HMMER3's hmmscan utility \citep{Eddy1995,Johnson2010,Eddy2011,Mistry2013}
        and sequences with a hit to a PFAM domain at an independent E-value of \(1e^{-5}\) were retained.
    \item These hits were then finally filtered for PFAM domains which mapped to transporter families classified by the Transporter Classification Database (TCDB) \citep{Saier2006,Saier2008,Saier2009,Saier2014}
        mapping files.
\end{enumerate}

Additionally, to ensure thorough discovery of 
all \textit{M. reisseri} transporters \textit{M. reisseri} binned sequences 
were BLASTP-ed against the NR protein database with an e-value of \(1e^{-3}\) and 20 hits.
InterproScan \citep{Zdobnov2001a} was then used to 
further annotate these proteins incorporating
results from BlasProDom \citep{Servant2002}, FPrintScan \citep{Attwood1994}, 
HMMER \citep{Eddy2001} scans against the PIR \citep{Barker1998}, PFAM \citep{Bateman2002}, 
SMART \citep{Schultz1998}, PANTHER \citep{Thomas2003a} and TIGRFAM databases \citep{Haft2003}, 
ProfileScan \citep{Gribskov1988},
HAMAP \citep{Lima2009}, PatterScan, 
SuperFamily \citep{Gough2002}, 
SignalP \citep{Petersen2011}, TMHMM \citep{Sonnhammer1998}, 
Gene3D \citep{Buchan2002}, Phobius \citep{Kall2007}
and Coils. Results were then mapped to GO terms \citep{Ashburner2000,Harris2004}
and annotated via BLAST2GO \citep{Conesa2005a}.

Finally, all proteins annotated to have a GO term associated with ``transport'' and
``transport activity'' specifically, GO:0005215, GO:0005478 and GO:0006810 and their child
terms were extracted.  These were added to the already identified transporter
proteins using TMHMM/TCDB pipeline.

\section{Results}

\subsection{Kodama Assembly}

\begin{table}
    \begin{tabular}
        \hline
        \textbf{Preprocessing} & \textbf{PE Reads} \\
        \hline
        \textbf{Raw Reads}  & \(2.323\cdot10^{8}\)\\
        \textbf{Q30 Trimmed} & \(1.75\cdot10^{8}\)\\
        \textbf{Q5 Trimmed}  & \(2.127\cdot10^{8} \) \\&
        \textbf{Q5 Error Corrected}  & \(2.021\cdot10^{8} \)\\
        \textbf{Digital Normalistion} & \(1.09 \cdot10^{7}\)\\ 
        \textbf{K-mer abundance filtering} & \(1.055\cdot10^{7}\)\\
        \hline
    \end{tabular}
    \caption{Summary of read pre-processing stages for tthe Kodama library}
    \label{tab:kodama_preproc}
\end{table}

\begin{table}
    \begin{tabular}
        \hline
        \textbf{Assembly} & \textbf{Contigs} & \textbf{Likelihood (\(-log\))\\
        \hline
        \textbf{Trinity Q5 Normalised}  & 101,957 & \(1.216\cdot10^9\)\\
        \textbf{Bridger Q5 Normalised} & 62,504 & \(1.285\cdot10^9\)\\
        \textbf{Trinity Q30} & 53,938  & \(5.619\cdot10^{9} \) \\
        \hline
    \end{tabular}
    \caption{Summary of Kodama assembliesj}
    \label{tab:kodama_assembly}
\end{table}

Therefore, the optimal assembly chosen for further analysis was the Trinity
Q5 normalised assembly on the basis of RSEM-EVAL score. 

From the 101,957 transcripts 193,906 ORFS were called using tetrahymena 
encoding and 20,875 universal.

These were subsequently binned using the same approach as used in 
Chapter 3. 

\begin{table}
    \begin{tabular}
        \hline
        \textbf{Bin} & \textbf{Number of Transcripts} \\
        \hline
        Food & 3,873 \\
        Endosymbiont & 8,627 \\
        Host & 53,295 \\
        Unknown & 36,162 \\
        \hline
    \end{tabular}
\end{table}

Finally, ``Host'' and ``Endosymbiont'' binned transcripts 
were re-ORF called using the appropriate encodings to result in a 
host ORF bin of  sequences
and an endosymbiont bin of 5,565 peptides. 

\subsection{Transporter Identification}

To identify transporters present in the translated protein dataset of the 
endosymbiont bins of the CCAP1660/12 and YADGN1 
\textit{Paramecium bursaria} transcriptome assemblies, as well as the Chlorella NC64A 
and Coccoymxa C-169 predicted proteomes the following process was used:

\begin{table}
    \begin{tabular}[|c|c|c|c|c|]
        \hline
        & \textit{C. variabilis 1N} & \textit{M. reisseri CCAP 1660/12} & \textit{C. vulgaris NC64A} & \textit{C. subellipsoidea C-169} \\
        \hline
        Peptides        & 5,565 & 4,275 & 9,791 & 9,629 \\
        1+ TM domains   & 695 & 419 & 1,722 & 1,709 \\
        1+ TM and TCDB  & 251 & 185 & 690 & 697 \\
        \hline
    \end{tabular}
\end{table}






A set of 233 
transporter proteins were also identified 
in the \textit{M. reisseri} binned peptides
by parsing the results of a 
BLAST/InterProScan and GO term based annotation pipeline.
Of, these 77 were redundant to proteins already identified using
the TM/TCDB pipeline, therefore 156 were novel.

This means a total of 341 transporters were identified 
belonging to the CCAP 1660/12 endosymbiont from the SCT transcriptomes. 


\begin{figure}
    \includegraphics[width=\textwidth]{endosymbiont_bin_annotation.pdf}
    \caption{}
    \label{BLAST hits at a more stringent threshold than 
    that used in the binning thus the lack of 100 annotation.}
\end{figure}

\subsection{Kallisto quantification analysis}

Using the nucleotide CDS sequence of the called peptides identified as transporters
from the primary SCT transcriptome. 

Due to the compositional/coverage biases of MDA-based single cell transcriptomics 
Kallisto statistical inference was likely to be spurious and relate to the 
well-documented coverage biases of MDA. Therefore, a simple presence/absence
filter was implemented for the single cell libraries where an estimated
Transcripts per million (TPM) was above 0 for at least 1 biological replicate
in each condition. 
TPM is an estimate of the 


\begin{figure}
    \includegraphics[width=\textwidth]{transporter_expression_heatmap.svg}
    \caption{A binary filter heatmap that displays the 4 groups of transporter
        proteins identified in the \textit{P. bursaria}-\textit{M. reisseri} 
        transcriptome.  Specifically, it shows 33 transporters expressed only
        in the dark single cell libraries, 54 expressed only in the light,
    84 expressed in both libraries and 13 only recovered in the bulk transcriptomes}
    \label{fig:binary_expression_heatmap}
\end{figure}


Of these only 2 were expressed in all 4 bulk libraries 34 in all 3 light libraries, 14 in all taxonomicall screened SCT libraries.

\begin{table}
    \begin{tabular}
        \hline
        \textbf{State} & \textbf{Name} & \textbf{TCDB Identity} \\
        \hline
        All Dark Libraries  & comp11781\_seq0|m.10145 & PF00448.18 & General Secretory Pathway \\
                            & comp20734\_seq0|m.20988 & PF00122.16 & P-type ATPase \\
        \hline
        All Light Libraries & comp34406\_seq1|m.34111 & PF02705.12 & K+ potassium transporter/Drug-Metabolite Transpoter/K+ uptake permease\\
                            & comp55761\_seq0|m.51120 & PF07690.12 & Major Facilitator Superfamily \\  % GPH:cation symporter ATP:ADP antiporter family -MFS CAN DO SUGAR
                            & comp26454\_seq0|m.27109 & PF03239.10 & Iron permease FTR1 family  \\
                            & comp12997\_seq0|m.11462 & PF02653.12 & Brnached-chain amino acid transport system/permease/ATP-binding cassette\\
                            & comp2196\_seq0|m.2436   & PF00361.16 & Proton-conducting membrane transporter, H+/Na+ translocating NADH dehydrogenase family, monovalent cation (K+/Na+): Proton antiporter-3 \\
                            & comp30376\_seq0|m.30648 & PF01490.14 & Transmembrane amino acid transporter, MFS/Amino acid/auxin permease\\
                            & comp8796\_seq0|m.7077   & PF01241.14 & Plant Photosystem 1 Supercomplex (psaG/psaK) \\
                            & comp16529\_seq0|m.15912 & PF00032.13, PF00033.15 & Cytochrome b (PRC) \\
                            & comp39264\_seq0|m.37815 & PF01970.12 & Tripartite tricarboxylate transporter (TctA) \\
                            & comp12686\_seq0|m.11025 & PF01970.12 & TctA family  \\
                            & comp18033\_seq0|m.17793 & PF03401.10 & TctC family \\
                            & comp16603\_seq1|m.16010 & PF00860.16 & Permease (CDF) \\
                            & comp1093\_seq1|m.1645   & PF00860.16 & Permease (CDF) \\
                            & comp8621\_seq0|m.6954   & PF00033.15 & Cytochrome b (PRC) \\
                            & comp23923\_seq0|m.24587 & PF00032.13 & PRC, Proton-translocation Quinol:Cytochrome c Reductase\\
                            & comp23290\_seq0|m.23888 & PF00115.16 & COX1 Cytochrome C and Quinol oxidase polypeptide I \\
                            & comp19868\_seq0|m.19974 & PF00115.16 & COX1 \\
                            & comp47698\_seq0|m.44756 & PF01061.20 & ABC-2 type transporter \\
                            & comp27137\_seq0|m.27822 & PF06472.11 & ABC-2 \\
                            & comp33855\_seq0|m.33598 & PF00528.18 & Binding-protein dependent transporter system inner membrane (ABC)\\
                            & comp38129\_seq0|m.36919 & PF00528.18 & Binding-protein dependent transporter system inner membrane (ABC)\\ 
                            & comp29161\_seq0|m.29630 & PF00528.18 & Binding-protein dependent transporter system inner membrane (ABC)\\
                            & comp30550\_seq0|m.30821 & PF02417.11 & Chromate ion transporter \\
                            & comp36383\_seq0|m.35530 & PF02487.13 & Equilibrative Nucleoside Transporter (ENT)\\
                            & comp2716\_seq1|m.2811   & PF01490.14 & Transmembrane amino acid transporter \\
                            & comp31515\_seq0|m.31652 & PF00146.17 & NADH dehydrogenase \\
                            & comp15817\_seq0|m.15012 & PF02990.12 & Endomembrane protein-70 \\
                            & comp23811\_seq0|m.24460 & PF03030.12 & \(H^{+}\)-\(Na^{+}\) translocating Pyrophosphatase Family \\
                            & comp12244\_seq0|m.10668 & PF03030.12 & \(H^{+}\)-\(Na^{+}\) translocating Pyrophosphatase Family \\
                            & comp17395\_seq0|m.16982 &  & ABC transporter \\ 
                            & comp38285\_seq0|m.37041 & & Leucine-Isoleucin valine transpoter \\
                            & comp61444\_seq0|m.55210 & & ABC transpoter \\
                            & comp12398\_seq0|m.10763 & & Cytochrome b6f \\
        \hline
        All SCT libraries   & comp23196\_seq0|m.23818 & PF00223.15 & PSI psaA/psaB \\
                            &  comp16798\_seq0|m.16234& PF00223.15 & PSI psaA/psaB \\
                            &  comp13220\_seq1|m.11682& PF00223.15 & PSI psaA/psaB \\
                            &  comp13220\_seq0|m.11681& PF00223.15 & PSI psaA/psaB \\ 
                            &  comp1772\_seq0|m.2188 & PF00421.15 & PSII \\
                            &  comp1772\_seq1|m.2192& PF00421.15 & PSII \\
                            &  comp2799\_seq0|m.2876 & PF00361.16, PF06455.7, PF00662.16 & Proton-conducting membrane transporter\\ %NADH + NADH-Ubiquonin oxioreductase
                            &  comp2799\_seq0|m.2878 & PF00361.16, PF06455.7, PF00662.16 & Proton-conducting membrane transporter\\ %NADH + NADH-Ubiquonin oxioreductase
                            &  comp2682\_seq0|m.2777& PF03911.12 & Sec61-beta \\
                            &  comp2682\_seq2|m.2790& PF03911.12 & Sec61-beta \\
                            &  comp6740\_seq0|m.5524& PF00119.16 & ATP synthase A \\
                            &  comp16837\_seq0|m.16280& PF00510.14 & COX3 \\
                            &  comp428\_seq0|m.891& PF01333.15 & QCR superfamily \\
                            &  comp15996\_seq1|m.15230 & & ABC Transporter \\
        \hline
    \end{tabular}
    \caption{A list of CDS identities that were predicted to be expressed in all the single cell
    libraries of a given type.}
    \label{tab:consensus_transporters}
\end{table}

After filtering photosystems and cytochrome related transporters = number left


2 homologs of HUP1,2,3 - m.12814 m.17059 (latter in b2go has no TM domain - partial ?)


Nitrate Reductase - m.43047  
NADH-glutatamte reductase dehyodrgenase ?
gluatmine synthetase ?
No high e-value hit for Nitrite reducatase

Ammonium transporter m.62060 in b2go not in TM pipe











\section{Discussion}

SEECER and K-mer abundance filtering may be mutually exclusive

\subsection{Quantification in MDA}

%Titrating ERCC spike-ins to level of single cell 
%It has become an increasingly established methodology to add RNA of known concentrations
%to extractions before library preparation to aid quantification
%and normalisation in downstream analysis. 
%Unforunately, there are no established methods for synthetic RNA spike-ins
%of known RNA concentrations (e.g. ERCC RNA standards \citep{Jiang2011}) with MDA
%sc-RNAseq. Specifically, standards would need to be carefully titrated to appropriate
%concentrations for single cell methods or they would overwhelm sequencing 


%therefore no such spike-ins were added. 

%    \item Just use SCT for transcript quantification i.e. map the SCT reads to a bulk derived \textit{de novo} assembly to generate counts (and is there a severe bias induced 
%        by the MDA amplification \citep{Liang2014}?)
%\end{itemize}  <>++
%However, MDA is prone to a degree of amplification bias \citep{Liang2014} which may be problematic in accurate inference of differential expression
%therefore, the suitability of MDA-based scRNA-Seq in particular will also need to be assessed.
%The only other published analysis of \textit{Paramecium bursaria} and its green algal endosymbionts by \citep{Kodama2014} largely
%side-stepped this issue by focussing on the analysis of host transcripts with and without the endosymbiont by filtering
%likely endosymbiont derived contigs from analysis using a crude MEGABLAST \(e^{-40}\) approach.
%Studies in related ciliates have demonstrated a high prevalence of alternative splicing events (5.2\% of genes in \textit{Tetrahymena
%    thermophila} for a single celled eukaryote \citep{Xiong2012}
%This paper also demonstrated the huge dynamic range of expression (and thus necessity of RNA-Seq over microarray approaches) in \textit{T. thermophila})
%with approximately 6 order of magnitude range \citep{Xiong2012}
%However, as both target species - host and endosymbiont are eukaryotes the complication of mRNA enrichment 
%is simplified due to the sufficiency of poly-A selection for this task (instead of rRNA depletion methods)
%Determining the necessary sequencing depth is also difficult.
%First achieved in \citep{Lao2009}





MDA has amplification bias with GC content - problem for PbMr \citep{Macaulay2014}


\subsection{Potentially missing transporters}

One potential issue with the binning methodology used is
that any recently horizontally acquired host or endosymbiont 
transporters or other metabolic pathway proteins will have been misclassified
and potentially discarded into the ``food'' or ``unknown'' bin.

This is problematic as there is evidence for bacterially acquired 
hexose-phosphate transporters playing a key role in the 
establishment of primary plastid endosymbiosis \citep{Price2012,Karkar2015a}.

Ideally, future work could expand this transporter analysis over the 
``unknown'' and ``food'' binned sequences in combination with synteny
analyses using genomic sequences to investigate and identify
potential horizontally acquired transporters that may play a role.

Another issue with the binning approach used is the possibility
of totally novel transporter (and other proteins) not being classified
due to the dependence of the binning on homology to known sequences.
Therefore, totally novel proteins would not have been identified as 
``host'' or ``endosymbiont''.  Unfortunately, this problem 
could only properly be resolved with a thorough and robust 
draft quality genome for both host and endosymbiont which was outside
the scope of this analysis. 

The final source of obfuscation in an accurate analysis of this data
is that of host-endosymbiont gene transfer.  
This is well observed phenomenom that has resulted in the eventual loss
of the endosymbiotic in the majority of algal secondary endosymbiotic organelles
as genes are transferred to the host nucleus
\citep{Keeling2008a,Archibald2005,Keeling2004}.

It is unknown and difficult to determine to what extent the 
unusual nuclear dimorphism and germline sequestration of the host 
effect's the rate of this form of transfer. 

Fortunately, this binning method means that while some peptides
may have been falsely assigned to wrong bin all ``host'' and
``endosymbiont'' ORFs that were not either so novel they lacked
any homology to known proteins or were recently acquired from
bacteria were still included in this analysis. 

However, as some \textit{M. reisseri} and \textit{Chlorella} endosymbionts 
have been demonstrated as capable of free-living it is unlikely
that HGT has occurred between host and endosymbbiont as extensively
as that observed in established photosynthetic organelles. 






\section{Conclusion}


Network analysis
http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3299011/pdf/fmicb-03-00085.pdf

