\graphicspath{{chapters/5.Chapter_3/figures for 

\begin{savequote}[75mm]
``All models are wrong, but some are useful''
\qauthor{- George E.P. Box & Draper: \textit{Empirical model-building and response surfaces, 1987}}
\end{savequote}

\chapter{Metabolic integration}

\section{Introduction}

The linking of metabolism between host and endosymbiont is a fundamental 
stage in endosymbiotic integration \citep{Bhattacharya2007,Karkar2015a}.
Complementation of respective metabolic deficiencies/limitations in
host and endosymbiont allow exploitation of novel niches and provide
the key selective benefits of endosymbiosis \citep{Hoffmeister2003}.

In order to identify points of metabolic integration it is necessary to
identify the primary ``points of contact'' between the metabolic networks 
of host and endosymbiont. These ``points of contact'' comprise two
major classes of proteins, transporters and secreted proteins. 
Specifically, host and endosymbiont transporters which localise to the
perialgal vacuole (PV) membrane and the outer-membrane of the endosymbiont,
and proteins which are secreted by host and endosymbiont into 
PV.  
The importance of these systems in metabolic integration is further 
evidenced by the the way the evolution of protein import systems 
from host to endosymbiont, such as the TICs/TOCs and TIMs/TOMs
of plastids and mitochondria, is considered by many 
to be the main constraint in the establishment of an endosymbiont
as an organelle \citep{Pfanner2001.Keeling2008a}.


These predictions can be directly supplemented using the rest of
host and endosymbiont classified transcripts in a metabolic mapping 
approach.  This involves annotations and
analysis of the presence and absence of metabolic pathway components.


Finally, metabolic integration can be investigated by the 
direct analysis of metabolites.  This has the benefits of not being

relying on transcriptomics using both
targeted and untargeted metabolomics approaches. 


Currently we know cations, amino acids, saccharides such as maltose


\subsection{Transporters}
The most important group of proteins in the control and evolution
of metabolic integration is that of host and endosymbiont transporter proteins.
This is due to their role in facilitating diffusion and active transport
across the lipid membranes that exist between host and endosymbiont.
While the role of transporters has been investigated extensively in 
several symbioses, these typically focus on organellar endosymbionts where
metabolic co-dependence has become fixed \citep{Yuan1994,Tyra2007,Huang2007,Li2010a} 
or ecological community interactions 
\citep{Richards2013,Hirner2006,Bachmann2013,Oldroyd2009}.


Without transporters many metabolically important 
large uncharged polar molecules (e.g. carbohydrates, amino acids)
and charged molecules (e.g. the various biologically relevant cations and anions such
as \(H^{+}\)) are incapable of significant rates of diffusion 
across membranes even in the presence of high concentration gradients. 
Therefore, the presence of transporters is vital to facilitating
any interaction involving these groups of metabolites. 
However, there are also several metabolites that are capable 
of unfacilitated diffusion across lipid membranes at significant rates.
These include important respiratory gases such as \(O_{2}\) and \(CO_2\),
hydrophobic compounds like benzene and small uncharged polar molecules
(e.g. \(H_2O\) and ethanol) \citep{cooper2013the,alberts2015molecular}.  
Despite this, transporter proteins have evolved to facilitate
even more rapid diffusion of some of these metabolites e.g. aquaporins
\citep{Agre1993}.  
Finally, transporters are important in that some can provide the ability
to actively transport metabolites against concentration gradients. 
This involves the expenditure of energy (typically in the form of ATP)
to directly pump compounds or generate an opposing gradient which can be
exploited (primary vs secondary active transport).


In the case of \textit{P. bursaria} and its endosymbiont we are
particularly interested in the host 
the host perialgal vacuole membrane and the outer membrane of the green
algal endosymbiont in the case of \textit{P. bursaria}.  
Therefore, the first step to the successful analysis of the metabolic
integration of host and endosymbiont is accurate identification of
transporter proteins present in their respective binned transcriptome
sequences.   By identifying both the identity of these proteins 
and qualitatively investigating the relative day:night expression
targets can be generated for further analysis, validation, and 
proof of localisation to the perialgal vacuole and algal outer
membrane.


Transporter proteins can be predicted 






A pilot untargeted global metabolomic profile was generated
for \textit{P. bursaria}-\textit{M. reissieri} CCAP 1660/12 
culture to test the feasability of this approach and help optimise
chromatography and 


Finally, a proof of concept for the targeted metabolomic
analysis of this system was conducted used HPLC-QQQ
to determine the relative abundances of amino acids
in the culture between day and night conditions. 



In the case of \textit{P. bursaria} and its green algal endosymbionts 
there are



Of particular interest in \textit{P. bursaria}-\textit{M. reisseri} 
are the transporters which play a role in the most obvious point of 
metabolic integration between a host
and a photosynthetic endosymbiont 


is that of the products of endosymbiosis. 
Specifically, the exchange of photosynthates in the form of various
carbohydrates and gas exchange. 

\subsubsection{Sugar Transporters}

Since their discovery \citep{Cohen1957}, several functional types of transporter proteins have been identified:
uniporters which facilitate transfer along the sugar gradient (e.g. hexose
transporter (HXT), glucose-transporter (GLUTs), and SWEETs),
and cotransporters which couple transport to another gradient. 
Co-transporters are further divided into symporters such as sodium-glucose symporters (SGLTs), sucrose-\(H^{+}\) 
cotransporter (STP), and SUT which co-transport cations in the same direction and
antiporters such as the tonoplast sugar transporter (TST) that uses opposing
gradients \citep{Chen2015}.


Polyol transporters?
Carrier proteins includfe uniporters and cotranspoters but not channels
Active transport involves transport using energy
Secondary active transport are cotransporters that exploit a primary active transport crated
gradient (i.e.pump)



The transfer of matlose, glucose, fructose and malate from endosymbiont
to host has been observed using radiolabelling 
\citep{Brown1974}.  Furthermore, green algae strains competent to
 form endosymbioses were found to inducibly release significantly
 more photosynthate (in the form of \(\sim 95\%\) maltose) than strictly free-living strains
 in the presence of \(NaHCO_3\) on the order of \(5.4-86.7\%\) vs. \(0.4-7.6\%\)
 of total photosynthate \citep{Muscatine1967a}.

 pH-dependent release of photosynthate \citep{}



There are 3 classes of eukaryotic sugar transpoters: major facilitiator superfamily (MFS) 
transporters including glucose
transporters (GLUTs), HXTs, STPs and SUTs; the Sodium-Solute Symport Family (SSF) sodium-glucose symporters (SGLTs); 
and SWEETs \citep{Chen2010a,Chen2015}.

Additionally, there are a few transporters mainly associated with intracellular transport such as 
vacuolar glucose transporter I (VGT1), ERD-like transporters and tonoplast monosaccharide transporters.


\textit{C. vulgaris} inducible system for active hexose uptake \citep{Tanner1974}






MFS typically have 12 TM domains, 

SWEETs are the smallest comprising only 7 TM domains \citep{Chen2015}

Typically, there are multiple isoforms with varied regulation, kinetics and specififity.




HXTs, GLUTs, STPs and SUTs are MFS
SGLT aren't



HUPs are a type of STPs \citep{Chen2015} and of particular interest due to their discovery in green
algae \citep{Sauer1989}




Additionally there is a large amount of internal transport 





\subsubsection{Nitrogen/Amino Acid Transporters}

Nitrite reductase activity stimulated by nitrated but no nitrate reductase
\citep{Kamako2005}




 Respiration of the host plays a role in the gas exchange of \(CO_2\) and \(O_2\) 
 to the endosymbiont, with the algae displaying higher levels of photosynthetic
 activity while in association with the host due to the increase host related 
 \(CO_2\) respiration \citep{Reisser1980}.

Addition of glucose, which likely increases host respiration rate and thus \(CO_2\)
evolution increases the rate of photosynthetic oxygen production commensurately \citep{Reisser1980}




Nitrogen is the ost transferred material after carbon \citep{Kato2009}, 
A huge range of states and parametetrs have been studied showing a range
of nitrate annd potentuially maino acid based systems (particularl L-glutamine and ammonium \citep{Albers1982}).


Neither green or algae free \textit{P. bursaria} were found to take up nitrate 
Ammonia is excreted by host without algae \citep{Albers1982}.


Proton gradient dependent transport of maltose/photosynthate out of the \citep{Schussler1992}
pH induces release of maltose in many algae \citep{}




These early results are unclear on specific strains and how they correspond
to modern taxa \citep{Kato2009}. 



%Transporters are necessary for the transport and localisation of effectors and metabolites
%that are too large to diffuse through the membrane and for active transportation
%against concentration gradients. 
%
%When considered from a purely quantitative and mass perspective, sugars
%are the most important organic compounds to be taken up by living organisms. 
%As well as their obvious role in energy processes 

%There are 4 archetypical 
%\subsection{Sugar Transporters}
%
%In \textit{Parachlorella kessleri} (previous \textit{Chlorella kessleri} \citep{Krienitz2004,Hoshina2010})
%
%Chlorella \(H^{+}\)-monosaccharide (HUP1) co-transporters \citep{Sauer1989}
%have been identified (via hexose-transport deficient mutant screens \citep{Sauer1986}
%and localised to the plasma membrane \citep{Stadler1995}
%
%
%Additionally these transporters have been identified in the
%\textit{Auxenochlorella protothecoides} (formerly \textit{Chlorella protothecoides} \citep{Champenois2014}) genome \citep{Gao2014}.
%
%\subsection{Amino acid transporters}



There are 3 previously established points of metabolic contact between 
\textit{P. bursaria} and its green algal endosymbionts. 

Namely, sugar transport (mainly maltose), 

By analysis of a F36-ZK \textit{C. variabilis} \citep{Hoshina2010} endosymbiont
that 


Other \textit{C. variabilis} strais 
and \textit{M. reisseri} have previously been found to utilise nitrate 
in the same manner has other plants and algae as a nitrogen
source by the conversion to ammonium by nitrate reductase (NR)
and nitrite reductase (NiR).


What do \textit{Coccomyxa} sp. and \textit{C. vulgaris} use?



Japanese \textit{C. variabilis} 



\subsection{Secreted proteins}

The next major class of proteins involved in endosymbiosis
from the perspective of the endosymbiont as those
which are secreted.   These proteins are secreted
into the perialgal vacuole and thus are likely to
be play some form of role in the maintenance of
the endosymbiosis.


Identification of secreted proteins also offers
a chance to further elucidate the function of novel or poorly 
characterised transporters that have been identified. 
For example, 

Secreted proteins can be effectively identified by 
searching for the presence of N-terminal signal peptides. 

These are short 15-30 amino acid N-terminal sequences cleaved during translocation.
They aren't conserved sequences but are present across the tree of life.

Feature a biochemical compositional fingerprint 
of positively charged residues, hydrophobic residues and polar uncharged
residues with distinctive points before site.l \citep{Emanuelsson2007}





Not all have signal peptides either





SignalP \citep{Nielsen1997} has proven the most effective method of predicting
the presence of signal peptides \citep{Lee2009a,Petersen2011}.


This method uses a standard feed-forward artificial neural network
with 8-41 hidden units (depending on whether the organism is eukaryotic, 
gram positive or gram negative) trained with back-propagation.




By analysing the identified signal peptides it is possible
to 






    








\subsection{Metabolic mapping}


Mapping 


\subsection{Metabolomics}

\subsubsection{Untargeted Global Profiling}
GCMS, LCMS,

\subsection{Targeted Analysis of Key Classes of Compounds}

Mass spectrometry also offers methods by which a targeted 
analysis of key classes of compounds can be conducted. 


While few

Amino acids 



\section{Aims}

The principal aim of this chapter is to identify
likely points of metabolic integration between host and 
endosymbiont to generate targets for subsequent 
targeted mass spectrometry, RNAi and qPCR based validation
experiments. 

This will be achieved by:
\begin{itemize}
\item identifying
    transporter proteins present in the endosymbiont
    binned transcripts from the CCAP1660/12 RNA-Seq analysis
    and analysing them for qualitative differential 
    expression between day and night. 
\item identifying
    secreted proteins present in the endosymbiont
    binned transcripts from the CCAP1660/12 RNA-Seq analysis
    and analysing them for qualitative differential 
    expression between day and night. 
\item Comparative analysis of metabolic pathways 
    between host and endosymbiont relative to 
    sequenced green algal genomes. 
\item A pilot untargeted global metabolomic profile of the 
    system and comparison of day to night.
\item Targeted quanitative analysis of amino acid concentrations
    between day and night. 
\end{itemize}

\section{Methods}
\subsection{Transporter Analysis} 
\subsubsection{Transporter identification pipeline}
Transporters were identified in the 4 sets of sequences (\textit{C. variabilis}, \textit{M. reisseri},
\textit{C. vulgaris} and \textit{C. subellipsoidea}) using the following set of pipelined filters:
\begin{enumerate}
    \item Transmembrane (TM) domains were predicted for each sequence using an HMM approach implemented as part of TMHMM2 \citep{Sonnhammer1998,Krogh2001}
and sequence predicted to contain at least 1 TM domain was extracted.
    \item These sequences were then used to search a PFAM database of profile HMMs \citep{Eddy1998} via HMMER3's hmmscan utility \citep{Eddy1995,Johnson2010,Eddy2011,Mistry2013}
        and sequences with a hit to a PFAM domain at an independent E-value of \(1e^{-5}\) were retained.
    \item These hits were then finally filtered for PFAM domains which mapped to transporter families classified by the Transporter Classification Database (TCDB) \citep{Saier2006,Saier2008,Saier2009,Saier2014}
        mapping files.
\end{enumerate}

Additionally, to ensure thorough discovery of 
all \textit{M. reisseri} transporters \textit{M. reisseri} binned sequences 
were BLASTP-ed against the NR protein database with an e-value of \(1e^{-3}\) and 20 hits.
InterproScan \citep{Zdobnov2001a} was then used to 
further annotate these proteins incorporating
results from BlasProDom \citep{Servant2002}, FPrintScan \citep{Attwood1994}, 
HMMER \citep{Eddy2001} scans against the PIR \citep{Barker1998}, PFAM \citep{Bateman2002}, 
SMART \citep{Schultz1998}, PANTHER \citep{Thomas2003a} and TIGRFAM databases \citep{Haft2003}, 
ProfileScan \citep{Gribskov1988},
HAMAP \citep{Lima2009}, PatterScan, 
SuperFamily \citep{Gough2002}, 
SignalP \citep{Petersen2011}, TMHMM \citep{Sonnhammer1998}, 
Gene3D \citep{Buchan2002}, Phobius \citep{Kall2007}
and Coils. Results were then mapped to GO terms \citep{Ashburner2000,Harris2004}
and annotated via BLAST2GO \citep{Conesa2005a}.

Finally, all proteins annotated to have a GO term associated with ``transport'' and
``transport activity'' specifically, GO:0005215, GO:0005478 and GO:0006810 and their child
terms were extracted.  

\subsubsection{Qualitative expression analysis}

Kallisto \citep{Bray2015} was used to pseudoalign and estimate abundances for 
all taxonomically screen single cell 
libraries (4 dark and 3 light) to the called ``endosymbiont''
binned CDS sequences from the \textit{P. bursaria} CCAP 1660/12
transcriptome (see Chapter 2).

Kallisto doesn't align reads to references in the same
manner as conventional short read alignment algorithms
such as Bowtie2 \citep{Langmead2012}.  Instead of specifically 
mapping a read to a set of co-ordinates it instead
determines which transcripts are compatible with the alignment 
of a given read.  This is achieved via decomposition of transcripts
in de-Bruijn graphs and fast k-mer hashing to compare reads to transcript
graph nodes in constant time.  These k-compatability classes are then used with
boostrapped expectation-maximisation to estimate transcript quantification and
determine uncertainty \citep{Bray2015}. 

Results were visualised and analysed using ``sleuth'' and the
seaborn plotting library \citep{michael_waskom_2015_19108}.

\subsection{Secretome prediction}

A conservative set of secreted proteins were predicted 
using the following pipeline:
\begin{enumerate}
    \item Signal peptides are detected using SignalP4.1 and mature sequences
        created for each sequence with a signal peptide
    \item The set of mature sequences was combined with the 
        
        
        sequences detected to have a TM domain using TMHMM is discarded
    \item Sequences with a signal peptide and no TM domain were then filtered
        using for those predicted as secreted by TargetP1.1 
    \item Finally these sequences were filtered down to those which had extracellular
        localisation in WoLFPSORT 0.2 
\end{enumerate}

Endosymbiont secreted proteins also had an additional filtering stage in which
any secreted protein which was found to have a Chloroplast targeting signal 
using ChloroP1.1 was removed.


\subsection{Metabolic mapping analysis}

First, predicted proteomes were obtained or generated for
\textit{Coccomyxa subellipsoidea} C-169, \textit{Chlorella variabilis}
NC64A, \textit{Chlorella variabilis} 1N, and \textit{M. reisseri}. 

For \textit{M. reisseri} the endosymbiont binned sequences from the transcriptomic
sequencing project discussed in the previous chapter were used. 
\textit{Coccomyxa subellipsoidea} C-169 genome project \citep{Blanc2012} version 2.0 
JGI annotated proteins (created 12-01-2014) were downloaded from JGI's
Phyotozome v10.3.1 \citep{Goodstein2012}. 
Similarly, the ``best'' annotated proteins from
version 1 of the \textit{Chlorella variabilis} NC64A genome project \citep{Blanc2010}
were downloaded from JGI's genome portal \citep{Grigoriev2011,Nordberg2014}

However, to obtain \textit{C. variabilis} 1N endosymbiont peptides 
a reassembly and binning of raw sequencing data from \citep{Kodama2014}
was conducted (discussed below). 

Once all sequences were acquired they were annotated using KEGG
orthology.  This was achieved 
using the KEGG Automatic Annotation Server (KAAS) \citep{Moriya2007a}
single-directional best hit with both BLAST and GHOSTZ \citep{Suzuki2014,Suzuki2015} 
method against the following 40 gene sets: \textit{Homo sapiens}, 
\textit{Drosophila melanogaster}, \textit{Caenorhabditis elegans},
\textit{Arabidopsis thaliana}, \textit{Gylcine max},
\textit{Vitis vinifera}, \textit{Oryza sativa}, 
\textit{Ostreococcous lucimarinus}, \textit{Ostreococcus tauri},
\textit{Micromonas} sp. RCC299, \textit{Cyanidioschyzon merolae},
\textit{Galdieria sulphuraria}, \textit{Saccharomyces cerevisiae},
\textit{Candida albicans}, \textit{Neurospora crassa}, \textit{Aspergillus nidulans},
\textit{Coccidioides immitis}, \textit{Schizosaccharomyces pombe},
\textit{Ustilago maydis}, \textit{Encephalitozoon cuniculi},
\textit{Monosiga brevicollis}, \textit{Dictyostelium discoideum}, 
\textit{Acanthamoeba castellanii}, \textit{Plasmodium falciparum} 3D7, 
\textit{Cryptosporidium hominis}, \textit{Tetrahymena thermophila},
\textit{Paramecium tetraurelia}, \textit{Phaeodactylum tricornutum},
\textit{Emiliania huxleyi}, and \textit{Guillardia theta}.

KEGG annotations were then plotted onto KEGG metabolic networks and compared 
to identify key aspects of differences between the algal species and
host and endosymbiont metabolic.

\subsubsection{\textit{Chlorella variabilis} 1 N assembly}
\(232.3M\) 100bp paired-end reads from \citep{Kodama2014}'s 
bulk RNAseq transcriptome of \textit{Paramecium bursaria} Yad1g (syngen
3, mating type 1) bearing \textit{Chlorella variabilis} 1N endosymbionts
were downloaded from the DNA Data Bank of Japan (DDBJ) \citep{Tateno2002,Kaminuma2011}
in Sequence Read Archive (SRA) format \citep{Leinonen2011,KodamaNRA2012b} (accession DRA000907 \citep{Kodama2014}).

These reads were then converted to fastq using ``fastq-dump'' using the SRA Toolkit
\citep{NationalCenterforBiotechnologyInformation2011}.  Reads were then trimmed
for sequencing adapters using ILLUMINACLIP and SLIDINGWINDOW with a window size
of 4 and a minimum average quality of 5 in Trimmomatic \citep{Bolger2014a}.

Reads were then error-corrected using ``SEECER'' with a k-mer size of 25 and 
default settings otherwise (entropy of 0.6 and a cluster log-likelihood
of -1) \citep{Le2013}.  Error-corrected reads were digitally normalised
using a K-mer size of 25 and a coverage of 20 \citep{Brown2012} and 
low abundance K-mers in high coverage reads were filtered \citep{Zhang2014,Zhang2015}
using the Khmer software package \citep{Doring2008,Crusoe2015}.

Assemblies were completed in a modified/fixed version of 
Bridger 2014-12-01 \citep{Chang2015} (available at
\url{https://github.com/fmaguire/Bridger_Assembler}) and 
Trinity v2.0.6 \citep{Grabherr2011,Haas2013} both with K-mer
sizes of 25.

An alternative Trinity assembly was also completed using
SLIDINGWINDOW Q30 trimmed reads without normalisation or 
error correction.

Assemblies were then compared using RSEM-EVAL \citep{Li2014} and the best
overall assembly selected on the basis of likelihood.

ORFs were called from the best assembly using universal and tetrahymena encodings 
via TransDecoder \citep{Haas2013} retaining the best scoring sequences and those
with HMMR hits to PFAM and BLASTP hits to the swissprot database. 

Phylogenies were generated for each sequence using the same approach and pipeline
described in Chapter 2. These phylogenies were subsequently classified using the 
same trained K-Neighbours supervised learning algorithm.
Any sequence that didn't have enough BLAST hits in the genomes used to generate
a phylogeny (5) were parsed based on what hits were retrieved.
Those with no hits were classified as ``unknown'' and those with
hits were classified based on the origin of those hits e.g. hits
to green algae and plant genomes were considered ``endosymbiont'' and so on.

Finally, the ORF bins for host and endosymbiont 
from both encodings were manually combined and reconciled
to generate transcript bins.  With the transcripts binned into
``host'' and ``endosymbiont'' ORFs were recalled from them using the appropriate
encodings. 


\subsection{Metabolomics}

\subsubsection{Untargeted Global Profiling}

3 mass spectrometry analyses were conducted to investigate the presence/absence
and relative abundances of 


\subsubsection{Targeted Amino Acids Quantitative Analysis}



\section{Results}
\subsection{Kodama Assembly}
\begin{table}
    \begin{tabular}
        \hline
        \textbf{Preprocessing} & \textbf{PE Reads} \\
        \hline
        \textbf{Raw Reads}  & \(2.323\cdot10^{8}\)\\
        \textbf{Q30 Trimmed} & \(1.75\cdot10^{8}\)\\
        \textbf{Q5 Trimmed}  & \(2.127\cdot10^{8} \) \\&
        \textbf{Q5 Error Corrected}  & \(2.021\cdot10^{8} \)\\
        \textbf{Q5 Digital Normalistion} & \(1.09 \cdot10^{7}\)\\ 
        \textbf{Q5 K-mer abundance filtering} & \(1.055\cdot10^{7}\)\\
        \hline
    \end{tabular}
    \caption{Summary of read pre-processing stages for tthe Kodama library demonstrating
    the massive amount of redundancy that digital normalisation removes from the assembly.
The low amount number of reads removed during K-mer abundance filtering indicates
that there were relatively few low abundance 



}
    \label{tab:kodama_preproc}
\end{table}

\begin{table}
    \begin{tabular}
        \hline
        \textbf{Assembly} & \textbf{Contigs} & \textbf{Likelihood (\(-log\))\\
        \hline
        \textbf{Trinity Q5 Normalised}  & 101,957 & \(1.216\cdot10^9\)\\
        \textbf{Bridger Q5 Normalised} & 62,504 & \(1.285\cdot10^9\)\\
        \textbf{Trinity Q30} & 53,938  & \(5.619\cdot10^{9} \) \\
        \hline
    \end{tabular}
    \caption{Summary of Kodama assembliesj}
    \label{tab:kodama_assembly}
\end{table}

Therefore, the optimal assembly chosen for further analysis was the Trinity
Q5 normalised assembly on the basis of RSEM-EVAL score. 

From the 101,957 transcripts 193,906 ORFS were called using tetrahymena 
encoding and 20,875 universal.

These were subsequently binned using the same approach as used in 
Chapter 3. 

\begin{table}
    \begin{tabular}
        \hline
        \textbf{Bin} & \textbf{Number of Transcripts} \\
        \hline
        Food & 3,873 \\
        Endosymbiont & 8,627 \\
        Host & 53,295 \\
        Unknown & 36,162 \\
        \hline
    \end{tabular}
\end{table}

Finally, ``Host'' and ``Endosymbiont'' binned transcripts 
were re-ORF called using the appropriate encodings to result in a 
host ORF bin of 61,239 sequences
and an endosymbiont bin of 5,565 peptides. 

\subsection{Transporter Identification}

To identify transporters present in the translated protein dataset of the 
endosymbiont bins of the CCAP1660/12 and YADGN1 
\textit{Paramecium bursaria} transcriptome assemblies, as well as the Chlorella NC64A 
and Coccoymxa C-169 predicted proteomes the following process was used:

\begin{table}
    \begin{tabular}[|c|c|c|c|c|]
        \hline
        & \textit{C. variabilis 1N} & \textit{M. reisseri CCAP 1660/12} & \textit{C. vulgaris NC64A} & \textit{C. subellipsoidea C-169} \\
        \hline
        Peptides        & 5,565 & 4,275 & 9,791 & 9,629 \\
        1+ TM domains   & 695 & 419 & 1,722 & 1,709 \\
        1+ TM and TCDB  & 251 & 185 & 690 & 697 \\
        \hline
    \end{tabular}
\end{table}






A set of 233 
transporter proteins were also identified 
in the \textit{M. reisseri} binned peptides
by parsing the results of a 
BLAST/InterProScan and GO term based annotation pipeline.
Of, these 77 were redundant to proteins already identified using
the TM/TCDB pipeline, therefore 156 were novel.

This means a total of 341 transporters were identified 
belonging to the CCAP 1660/12 endosymbiont from the SCT transcriptomes. 







\begin{figure}
    \includegraphics[width=\textwidth]{endosymbiont_bin_annotation.pdf}
    \caption{}
    \label{BLAST hits at a more stringent threshold than 
    that used in the binning thus the lack of 100 annotation.}
\end{figure}

\subsection{Kallisto quantification analysis}

Using the nucleotide CDS sequence of the called peptides identified as transporters
from the primary SCT transcriptome. 

Due to the compositional/coverage biases of MDA-based single cell transcriptomics 
Kallisto statistical inference was likely to be spurious and relate to the 
well-documented coverage biases of MDA. Therefore, a simple presence/absence
filter was implemented for the single cell libraries where an estimated
Transcripts per million (TPM) was above 0 for at least 1 biological replicate
in each condition. 
TPM is an estimate of the 


\begin{figure}
    \includegraphics[width=\textwidth]{transporter_expression_heatmap.svg}
    \caption{A binary filter heatmap that displays the 4 groups of transporter
        proteins identified in the \textit{P. bursaria}-\textit{M. reisseri} 
        transcriptome.  Specifically, it shows 33 transporters expressed only
        in the dark single cell libraries, 54 expressed only in the light,
    84 expressed in both libraries and 13 only recovered in the bulk transcriptomes}
    \label{fig:binary_expression_heatmap}
\end{figure}


Of these only 2 were expressed in all 4 bulk libraries 34 in all 3 light libraries, 14 in all taxonomicall screened SCT libraries.

\begin{table}
    \begin{tabular}
        \hline
        \textbf{State} & \textbf{Name} & \textbf{TCDB Identity} \\
        \hline
        All Dark Libraries  & comp11781\_seq0|m.10145 & PF00448.18 & General Secretory Pathway \\
                            & comp20734\_seq0|m.20988 & PF00122.16 & P-type ATPase \\
        \hline
        All Light Libraries & comp34406\_seq1|m.34111 & PF02705.12 & K+ potassium transporter/Drug-Metabolite Transpoter/K+ uptake permease\\
                            & comp55761\_seq0|m.51120 & PF07690.12 & Major Facilitator Superfamily \\  % GPH:cation symporter ATP:ADP antiporter family -MFS CAN DO SUGAR
                            & comp26454\_seq0|m.27109 & PF03239.10 & Iron permease FTR1 family  \\
                            & comp12997\_seq0|m.11462 & PF02653.12 & Brnached-chain amino acid transport system/permease/ATP-binding cassette\\
                            & comp2196\_seq0|m.2436   & PF00361.16 & Proton-conducting membrane transporter, H+/Na+ translocating NADH dehydrogenase family, monovalent cation (K+/Na+): Proton antiporter-3 \\
                            & comp30376\_seq0|m.30648 & PF01490.14 & Transmembrane amino acid transporter, MFS/Amino acid/auxin permease\\
                            & comp8796\_seq0|m.7077   & PF01241.14 & Plant Photosystem 1 Supercomplex (psaG/psaK) \\
                            & comp16529\_seq0|m.15912 & PF00032.13, PF00033.15 & Cytochrome b (PRC) \\
                            & comp39264\_seq0|m.37815 & PF01970.12 & Tripartite tricarboxylate transporter (TctA) \\
                            & comp12686\_seq0|m.11025 & PF01970.12 & TctA family  \\
                            & comp18033\_seq0|m.17793 & PF03401.10 & TctC family \\
                            & comp16603\_seq1|m.16010 & PF00860.16 & Permease (CDF) \\
                            & comp1093\_seq1|m.1645   & PF00860.16 & Permease (CDF) \\
                            & comp8621\_seq0|m.6954   & PF00033.15 & Cytochrome b (PRC) \\
                            & comp23923\_seq0|m.24587 & PF00032.13 & PRC, Proton-translocation Quinol:Cytochrome c Reductase\\
                            & comp23290\_seq0|m.23888 & PF00115.16 & COX1 Cytochrome C and Quinol oxidase polypeptide I \\
                            & comp19868\_seq0|m.19974 & PF00115.16 & COX1 \\
                            & comp47698\_seq0|m.44756 & PF01061.20 & ABC-2 type transporter \\
                            & comp27137\_seq0|m.27822 & PF06472.11 & ABC-2 \\
                            & comp33855\_seq0|m.33598 & PF00528.18 & Binding-protein dependent transporter system inner membrane (ABC)\\
                            & comp38129\_seq0|m.36919 & PF00528.18 & Binding-protein dependent transporter system inner membrane (ABC)\\ 
                            & comp29161\_seq0|m.29630 & PF00528.18 & Binding-protein dependent transporter system inner membrane (ABC)\\
                            & comp30550\_seq0|m.30821 & PF02417.11 & Chromate ion transporter \\
                            & comp36383\_seq0|m.35530 & PF02487.13 & Equilibrative Nucleoside Transporter (ENT)\\
                            & comp2716\_seq1|m.2811   & PF01490.14 & Transmembrane amino acid transporter \\
                            & comp31515\_seq0|m.31652 & PF00146.17 & NADH dehydrogenase \\
                            & comp15817\_seq0|m.15012 & PF02990.12 & Endomembrane protein-70 \\
                            & comp23811\_seq0|m.24460 & PF03030.12 & \(H^{+}\)-\(Na^{+}\) translocating Pyrophosphatase Family \\
                            & comp12244\_seq0|m.10668 & PF03030.12 & \(H^{+}\)-\(Na^{+}\) translocating Pyrophosphatase Family \\
                            & comp17395\_seq0|m.16982 &  & ABC transporter \\ 
                            & comp38285\_seq0|m.37041 & & Leucine-Isoleucin valine transpoter \\
                            & comp61444\_seq0|m.55210 & & ABC transpoter \\
                            & comp12398\_seq0|m.10763 & & Cytochrome b6f \\
        \hline
        All SCT libraries   & comp23196\_seq0|m.23818 & PF00223.15 & PSI psaA/psaB \\
                            &  comp16798\_seq0|m.16234& PF00223.15 & PSI psaA/psaB \\
                            &  comp13220\_seq1|m.11682& PF00223.15 & PSI psaA/psaB \\
                            &  comp13220\_seq0|m.11681& PF00223.15 & PSI psaA/psaB \\ 
                            &  comp1772\_seq0|m.2188 & PF00421.15 & PSII \\
                            &  comp1772\_seq1|m.2192& PF00421.15 & PSII \\
                            &  comp2799\_seq0|m.2876 & PF00361.16, PF06455.7, PF00662.16 & Proton-conducting membrane transporter\\ %NADH + NADH-Ubiquonin oxioreductase
                            &  comp2799\_seq0|m.2878 & PF00361.16, PF06455.7, PF00662.16 & Proton-conducting membrane transporter\\ %NADH + NADH-Ubiquonin oxioreductase
                            &  comp2682\_seq0|m.2777& PF03911.12 & Sec61-beta \\
                            &  comp2682\_seq2|m.2790& PF03911.12 & Sec61-beta \\
                            &  comp6740\_seq0|m.5524& PF00119.16 & ATP synthase A \\
                            &  comp16837\_seq0|m.16280& PF00510.14 & COX3 \\
                            &  comp428\_seq0|m.891& PF01333.15 & QCR superfamily \\
                            &  comp15996\_seq1|m.15230 & & ABC Transporter \\
        \hline
    \end{tabular}
    \caption{A list of CDS identities that were predicted to be expressed in all the single cell
    libraries of a given type.}
    \label{tab:consensus_transporters}
\end{table}

After filtering photosystems and cytochrome related transporters = number left


2 homologs of HUP1,2,3 - m.12814 m.17059 (latter in b2go has no TM domain - partial ?)


Nitrate Reductase - m.43047  
NADH-glutatamte reductase dehyodrgenase ?
gluatmine synthetase ?
No high e-value hit for Nitrite reducatase

Ammonium transporter m.62060 in b2go not in TM pipe






\subsection{Secreted proteins}

This resulted in a set of 44 proteins for endosymbiont 




One of the most intriguing of this set is that
of a putative raffinose synthase


comp65133\_seq0|m.57547 Alpha amylase catalytic domain family
1.38e-10
homologous to Raffinose synthase or seed imbibition protein Sip1


low number of reads mapping in 1 day and 1 night library

Constitutive expression in both bulk samples 




\subsection{Metabolomics} 

\subsubsection{Global profiling}

\begin{figure}
    \includegraphics[width=\textwidth]{metabolome_mds.pdf}
    \caption{Non-metric Multidimensional scaling of NBMS samples}
    \label{fig:metabolome_nmds}
\end{figure}


\begin{figure}
    \includegraphics[width=\textwidth]{gcms_cloud_plots.pdf}
    \caption{Cloud point showing for the GCQTOF analysis. 
    The radius of a given point reflected its fold change.
    Data is filtered to those 50 points with significantly 
different expression (FDR corrected P-value of 0.01 shown by 
depth of colour)}
    \label{}
\end{figure}

\begin{figure}
    \includegraphics[width=\textwidth]{lcms_cloud_plots.pdf}
    \caption{A: Positive 254,  B: 43}
    \label{}
\end{figure}


\begin{figure}
    \includegraphics[width=\textwidth]
    \caption{}
    \label{fig}
\end{figure}



Again one interesting peak is that of 
in negative
9.7 0.04846 DOWN with a fold vlaue of 9.7
putatively raffinose>? 503.1698  vs actual 503.169
    

\subsubsection{Targetet amino acid analysis}












\section{Discussion}

SEECER and K-mer abundance filtering may be mutually exclusive

\subsection{Quantification in MDA}

%Titrating ERCC spike-ins to level of single cell 
%It has become an increasingly established methodology to add RNA of known concentrations
%to extractions before library preparation to aid quantification
%and normalisation in downstream analysis. 
%Unforunately, there are no established methods for synthetic RNA spike-ins
%of known RNA concentrations (e.g. ERCC RNA standards \citep{Jiang2011}) with MDA
%sc-RNAseq. Specifically, standards would need to be carefully titrated to appropriate
%concentrations for single cell methods or they would overwhelm sequencing 


%therefore no such spike-ins were added. 

%    \item Just use SCT for transcript quantification i.e. map the SCT reads to a bulk derived \textit{de novo} assembly to generate counts (and is there a severe bias induced 
%        by the MDA amplification \citep{Liang2014}?)
%\end{itemize}  <>++
%However, MDA is prone to a degree of amplification bias \citep{Liang2014} which may be problematic in accurate inference of differential expression
%therefore, the suitability of MDA-based scRNA-Seq in particular will also need to be assessed.
%The only other published analysis of \textit{Paramecium bursaria} and its green algal endosymbionts by \citep{Kodama2014} largely
%side-stepped this issue by focussing on the analysis of host transcripts with and without the endosymbiont by filtering
%likely endosymbiont derived contigs from analysis using a crude MEGABLAST \(e^{-40}\) approach.
%Studies in related ciliates have demonstrated a high prevalence of alternative splicing events (5.2\% of genes in \textit{Tetrahymena
%    thermophila} for a single celled eukaryote \citep{Xiong2012}
%This paper also demonstrated the huge dynamic range of expression (and thus necessity of RNA-Seq over microarray approaches) in \textit{T. thermophila})
%with approximately 6 order of magnitude range \citep{Xiong2012}
%However, as both target species - host and endosymbiont are eukaryotes the complication of mRNA enrichment 
%is simplified due to the sufficiency of poly-A selection for this task (instead of rRNA depletion methods)
%Determining the necessary sequencing depth is also difficult.
%First achieved in \citep{Lao2009}





MDA has amplification bias with GC content - problem for PbMr \citep{Macaulay2014}

\subsection{What the heck is raffinose doing?}

galactinol + sucrose â†’ myo-inositol + raffinos/



Unfortunately, the GCQTOF analysis failed to successfully
identify sugars in the 


Raffinose has been associated with cold shock in
\textit{Parachlorella kessleri} (formerly \textit{C. vulgaris}),
accumulating duing cold exposure and disappearing after return
to normal culture temperatures 

It is unclear if raffinose is a 
\citep{Salerno1989}


Raffinose and another Raffinose Family Oligosaccharide (RFO) 
stachyose are generated from sucosrse in gymnopserls during
the cold season \citep{Kandler1982}.

Raffinose has been found to inhibit growth under
isosomotic conditions in a \textit{C. vulgaris} 
\citep{Setter1979}.

Specifically, it has also been directly associated
with cryoprotection of thykaloid membranes \citep{Lineberger1980}.
And no chlorop targeting sequence though

But that doens't explain secretion.



Bacterial raffinose transport system?


In archaeplastida RFOs are related to stroage
and transport of carbon.  



What about trehalose?
sucrose?




phylogeny?





It does appear that the raffinose synthase is constitutively
expressed unless both light and dark conditions are exposed to cold stress.

Culture temperature was \(18\celsius\), so this is unlikely. 

Needs confirmed with qPCR. 








\subsection{False positives}

There are a range of known sugar sensors that are non-functional
transporters derived from them - could have identified sensors as 
transporters \citep{Lalonde1999,Bianchi2010}

Therefore, it is key that the functionally important 

blah has to be tested

\subsection{Potentially missing transporters}

One potential issue with the binning methodology used is
that any recently horizontally acquired host or endosymbiont 
transporters or other metabolic pathway proteins will have been misclassified
and potentially discarded into the ``food'' or ``unknown'' bin.

This is problematic as there is evidence for bacterially acquired 
hexose-phosphate transporters playing a key role in the 
establishment of primary plastid endosymbiosis \citep{Price2012,Karkar2015a}.

Ideally, future work could expand this transporter analysis over the 
``unknown'' and ``food'' binned sequences in combination with synteny
analyses using genomic sequences to investigate and identify
potential horizontally acquired transporters that may play a role.

Another issue with the binning approach used is the possibility
of totally novel transporter (and other proteins) not being classified
due to the dependence of the binning on homology to known sequences.
Therefore, totally novel proteins would not have been identified as 
``host'' or ``endosymbiont''.  Unfortunately, this problem 
could only properly be resolved with a thorough and robust 
draft quality genome for both host and endosymbiont which was outside
the scope of this analysis. 

The final source of obfuscation in an accurate analysis of this data
is that of host-endosymbiont gene transfer.  
This is well observed phenomenom that has resulted in the eventual loss
of the endosymbiotic in the majority of algal secondary endosymbiotic organelles
as genes are transferred to the host nucleus
\citep{Keeling2008a,Archibald2005,Keeling2004}.

It is unknown and difficult to determine to what extent the 
unusual nuclear dimorphism and germline sequestration of the host 
effect's the rate of this form of transfer. 

Fortunately, this binning method means that while some peptides
may have been falsely assigned to wrong bin all ``host'' and
``endosymbiont'' ORFs that were not either so novel they lacked
any homology to known proteins or were recently acquired from
bacteria were still included in this analysis. 

However, as some \textit{M. reisseri} and \textit{Chlorella} endosymbionts 
have been demonstrated as capable of free-living it is unlikely
that HGT has occurred between host and endosymbbiont as extensively
as that observed in established photosynthetic organelles. 



\subsection{Why not further integration?}

Why is there not evidence of tighter integration in this system.


Firstly, the protein import systems considered necessary for extensive
EGT to start taking place are complicated
in the cases of secondary and tertiary endosymbioses
than basic plsatids due to the increased number
of membranes that may need to be traversed especially
for import directly to the secondary plastid from the
host \citep{Hirakawa2012}.


Secondly, the unusual nuclear dimorphism of the host \textit{P. bursaria}
may prove a barrier to the vast majority of EGT activity. 

For successful transfer to take place between host and endosymbiont it
would be necessary for the gene to transfer not just from the 
endosymbiont to the transcriptionally active host MAC but to the germline
MIC.  Even then integration into the MIC would have to occur in such a way
that it would be correctly spliced and duplicated during the conversion of the MIC
back to the MAC.
Compounding this with sexual reproduction further decreases the probability of
effective integration.



It should be noted that the prototypical hosts of the endosymbiotically ``promiscuous''
green algae - \textit{Chlorella}, \textit{Coccomyxa} and \textit{Micractinium} 
all display germline sequestration either through the aforementioned dimorphism
in \textit{P. bursaria} or via standard metazoan germlines in the case of
\textit{Hydra} and the kleptoplastic sacoglossan sea slugs. 





\section{Conclusion}


Network analysis
http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3299011/pdf/fmicb-03-00085.pdf

