%methods chapter
% Enough for an expert to reproduce without needing extensive references
% all data should be in thesis so they can be checked (appendix maybe and supplements)
% Methods commonly used throughout maybe leaving chapter specific methods to that chapter
\graphicspath{{chapters/2.Methods/figures}}

\chapter{Methods}

Materials and methods section

\begin{itemize}
    \item Culturing - how to culture paramecium/micractinium
    \item cDNA synthesis - creating cDNA from cultures, washing, cell picking etc
    \item Sequencing - general sequencing background sanger -> 2nd gen
    \item Genomics/transcriptomics - how this is specifically applied via HiSeq
    \item Phylogenetics - general, alignment, masking, model selection, tree reconstructions 
    \item Dendrogenous - UML of database, flowchart of program, comparison of parallelisation, profiling of tools process-level parallelism vs built in parallellism
\end{itemize}

univariate mathematical distributions reference\citep{Leemis2008}

%Might move to methods:\{
%flowchart of program tree-pipe
%
%UML of lab MYSQL Database
%
%Figure of serial tree pipe vs by-stage-parallel vs by-class parallel:
%e.g.
%     |XYZ                   X=1st stage of pipe, Y=2nd, Z=3rd etc .=syswait
%trees|   XYZ
%     |      XXYYZ             3rd form (by-class) is best because 2nd IO bounds
%     |
%     |X.Y.Z
%     |X.Y.Z
%     |XXYYZ
%     |
%     |XYZ
%     |XYZ
%     |XYYZZ
%     |____________
%        TIME
%
%
%Figure of profiling on multicore tools being quicker on random sample of dataset
%when they are run on a single core but multiple instances are run of them e.g.
%on 4 samples its quicker to run a 4-core tool as 4x1core instances than 4x4core
%\}
%
%

DRAFT DUE: END OF FEB



 
For reference RNA-Seq there is diminishing returns with 1M reads providing approximately the same level of accuracy in estimate transcript abundance as >30M reads for highly-expressed gene in six standard model organisms \citep{Lei2014} however for \textit{de-Novo} efforts like ours who knows



The correlation of transcript level to protein level is difficult


Benchmarks for mapping tools - include benchmarks paper showing we used the best tools generally or tested all the tools in the case of trinity etc
