\subsubsection{Extracting Nucleic Acids}

The first stage of any genomic or transcriptomic analysis is the extraction
of nucleic acids from the sample of interest.
Typically, this involves obtaining large amounts of source biomass either by
taking large environmental samples and fractionating using methods like filtration
and/or flow cytometry, or more typically establishing
a culture of the organisms of interest (if possible) and growing that up to sufficient biomass.
This secondary method does pose the problem of preventing the study of organisms
which cannot be cultured. 

(and with all the biases it involves) For bulk analyses we used a trizol based extraction
%
%For single cell gDNA CTAB \citep{}
%
%material either from a large
%environmental sample or by culturing 
%
%SPADES PAPER HAS GOOD CITES
While the quality of the output of earlier sequencing techniques
was highly reliant on both the quantity and quality of the input DNA (or cDNA) (not to mention
characteristics such as GC\%) precluding many systems from easy analysis.  This is especially
problematic for the majority of systems that are not easily culturable and thus can't be
grown up in sufficient quantities to easily extract sufficient nucleic acids to conduct.

Starting with single-cell single-gene sequencing experiments (e.g. \citep{Kuppers1993}) 
single cell methods allows obtaining data directly from individual cells
avoiding the bias and complication of culturing or having to decompose a complex environmental
metagenome or metatranscriptome containing a plethora of diverse cells \citep{Blainey2013}.  It also allows 
investigation of cells at particular life stages e.g. sexual reproduction without difficult
and potentially biasing synchronisation methodologies and the sequencing of long genes hard to obtain
from metagenomes.  

Single-cell genomic (SCG) sequencing has been demonstrated across the tree-of-life as sufficient to infer 
cellular metabolism \citep{Chitsaz2011}, the dynamics of interactions between organisms \citep{Yoon2011} and
environmental diversity \citep{Swan2013,Rinke2013}.

Likewise, single-cell transcriptomic (SCT) sequencing has been demonstrated to be an effective tool to investigate



%Bulk transcriptomic methods involve analysis of an ensemble of transcripts therefore expression 
%inferences are an average across all cells \citep{Stegle2015}
%Sufficient for some purposes, e.g. broad-stroke comparisons it masks the underlying and potentially
%biologically informative stochastic nature of gene expression \citep{Raj2008}.  SCT 
%revolutionises being able to compare different tissues, populations, and cell states in both
%single cell and multicellular systems by allowing direct interrogation of inter- and intra-
%grouping variation and could potentially revolutionise areas such as medicine \citep{Sandberg2014}
%and ecology (Moore Marine Microbe Project \url{http://marinemicroeukaryotes.org/}).
%While single cell level analyses have been possible for a while they have largely been restricted
%to low-throughput methods e.g. reporter constructs, FISH etc.
%
%Cell isolation, FACS, microfluidics, cell picking
%SCT has relatively high levels of technical noise specifically sampling noise for low-level
%expression and sample-specific variability in sequencing efficiency for highly-expressed 
%transcripts for one SCT method \citep{Grun2014}.
%
%
%mRNA was selected using a poly-A selection method before being fragmented, cleaned-up using ethanol and then 
%reverse transcribed into single-stranded cDNA using random hexamer primers.
%
%Currently, single cell genomics still requires whole genome amplification (WGA) even with 
%3rd generation sequencing platforms due to the relatively inefficient library preparation
%from a single cell (on the range of nanograms). \citep{Blainey2013} %%find support quote for transcriptomics needing WGA too
% 
%Multiple-displacement amplification \citep{Dean2001} is the main method for
%single-cell sequencing.
%REASONS IT IS GREAT
%However, it also generates many biases orders of magnitude differences in coverage
%in different regions 
%SINGLE CELL TRANSCRIPTOMICS ARE SUSPECT
%potential chimeric reads and incongruent read-pairs \citep{Bankevich2012}.
%
%This complicates assembly RODRIQUE PAPER MENTIONED IN SPADES 
%
%E+VSCV paper chisatz 
%
%However, such a different type of data that a new algorithm was needed not just modificaitons \citep{Bankevich2012}
%
%Paired-end data typically only used post-processing step of de-Bruijn graphs.
%Proper use of pairing information at the graph assembly phase is a potential 
%source of improvement.
%2011a medvedev PDBG but fixed insert size
%
%Idury and waterman DBG for assembly: originally considered infeasible as low coverage
%sanger even qitrh few errors needs error correction
%
%Pezner A-bruijn graphs
%everything else special case
%
%
%Single cell methods use multiple-displacement amplification to increase the concentraion of
%nucleic acids 
%
%For example. the Qiagen Repli-G kit used for both single cell transcriptomic and genomic analysis
%of the \textit{PbMr} system.
%
%Transcripts are reverse transcribed and then ligated into larger fragments.  MDA
%is then undergone on the larger fragments before the usual library preparation process.
%
%One interesting consideration for single cell transcriptomics is the possible generation
%of chimeric reads that will cross the boundary between two randomly ligated transcripts.
%
%However, the risk of this can largely be mitigated by selection of a relatively small fragment
%size for paired end sequencing.
%
%5'-3' - 5'-3'  - sequencing primers are unlikely to rpim

%\subsubsection{Annotation}
%
%Reed \textit{et al.} \citep{Reed2006} divided components annotation into a useful dimension paradigm as follows:
%\begin{enumerate}
%    \item Enumeration: identification of genes and assignment of their predicted or known functions. What?
%    \item Interaction: integration of protein-protein, regulatory and metabolic interaction data between components Can they interact?
%    \item Genomic Localisation: analysis of genomic localisation of genes i.e. epigenetics.
%    \item Plasticity: investigation of the change of other dimensions over evolutionary time. Does their interaction change over time? \citep{Reed2006} 
%\end{enumerate}
%
%Clearly, at a systems level it is difficult (if not impossible currently), 
%even in well defined model organisms, to complete a full 4-dimensional annotation 
%for all cellular components. However, it is more than feasible to have a largely
%complete 1D and even 2D annotation with 3D and 4D investigated for specific 
%components.  Furthermore, annotation can be conducted with a range of confidences
%from in silico predicted interactions to biochemically validating those predictions in vivo
%and incorporating the precision and recall of various methodologies. Generally, such a process
%is inherently iterative with successive model building and experimental testing and validation 
%or rejection of such models \citep{Reed2006}.
%
%One of the major goals of this thesis is to reconstruct a preliminary predicted
%2D annotation of the PbMr system.
%
%The most important step towards this goal is that of eukaryotic gene prediction.
%
%Eukaryotic gene prediction is achieved by generally two methods - \textit{de novo} HMM based
%methodologies (e.g. GENESCAN, TWINSCAN \citep{}<++>) and expression evidence based methods
%in which reads from sequenced cDNA (or in older approaches ESTs) are aligned to the 
%assembled genomic contigs \citep{Brent2007}.  Unfortunately even relatively deep
%sequencing of cDNA can fail to identify the structure of 20-40\% of genes in a typical
%eukaryote genome owing to these genes being expressed only at very low levels or in
%different conditions than that from which the cDNA was extracted \citep{Brent2007}.
%
%
%
%\subsection{Transcriptomics}
%While generally computationally simpler than genomic assembly \citep{MacManes2014}
%owing to the relatively fragmented nature of transcripts compared to genomic 
%sequences (i.e. lots of transcripts of various lengths rather than several 
%long chromosomes).  There a few key challenges to transcriptomic assembly
%that is absent in genomic work specifically:
%\begin{itemize}
%    \item Handling assembly of alternative isoforms of transcripts \citep{Pyrkosz2013}
%    \item Assembly of transcripts from gene-dense genomes with overlapping 5' UTR 
%    \item Assembly of data with highly heterogeneous read coverage, as transcript
%        expression level is proportional to read coverage (indeed expression analysis
%        using RNA-Seq datasets relies upon this fact).
%    \item Random hexamer bias
%    \item GC content bias, particularly so in PbMr
%\end{itemize}
%
%These problems are particularly problematic in \textit{de novo} approaches,
%however referenced assembly has its own issues in transcriptomics
%
%
%
%The correlation of transcript level to protein level is difficult
%
%Unfortunately, conventional RNA-seq requires either an established moderate-to-high
%density culture of the organism of interest 
%http://rnaseq.uoregon.edu/
%http://bioinformatics.bc.edu/marthlab/scotty/scotty.php
%
%Experiemntal design, read depth, ENCODE, Tarazona S 2012 fdiff exp in RNA-seq
%a matter of depth
%
%\begin{itemize}
%    \item Check Quality (FASTQC)
%    \item Trim reads (Trimmomatic)
%    \item Re-check quality (FASTQC)
%    \item Error-correct reads (?)
%    \item Normalize coverage (khmer)
%    \item Test paramters for multiple assemblers
%    \item Run each assembler with best parameters
%    \item Analyse all and merge best assemblies
%    \item Annotate transcriptome
%    \item Map reads to assembly, count
%    \item Differential expression analysis
%\end{itemize}
%
%
%
%
%Further complications from single cell:
%Fusions are created in cDNA ligation step prior to MDA in WTA
%However, poly-A tails should mean fusion reads are unlikely as they need to a
%span poly-A trac in all 4 possible fusions apart from 3'-5':5'-3'
%So to get high levels of chimeric reads we would need the two same transcripts
ombining
%

