\graphicspath{{appendices/appendix_1/figures}}
\chapter{Appendix 1}

\section{Arboretum classifier comparison}

\subsection{Genomes Used}
\begin{table}
	\begin{tabular}{|c|}
		\hline 
		\textbf{Genomes used in transcript binning}\\
		\hline
		\textit{Arabidopsis thaliana} \\
		\textit{Chlamydomonas reinhardtii} \\
		\textit{Ostreococcus tauri} \\
		\textit{Micromonas pusilla CCMP1545} \\  
		\textit{Chlorella variabilis NC64A} \\
		\textit{Chlorella vulgaris C-169} \\
		\textit{Physcomitrella patens} \\
		\textit{Saccharomyces cerevisiae S288C} \\ 
		\textit{Neurospora crassa OR74A}\\
		\textit{Homo sapiens}\\
		\textit{Mus musculus}\\
		\textit{Dictyostelium discoideum}\\
		\textit{Paramecium caudatum}\\
		\textit{Paramecium tetraurelia}\\
		\textit{Tetrahymena thermophila macronucleus}\\
		\textit{Oxytricha trifallax}\\
		\textit{Toxoplasma gondii}\\
		\textit{Guillardia theta}\\
		\textit{Bigelowiella natans}\\
		\textit{Emiliania huxleyi CCMP1516}\\
		\textit{Aureococcus anophagefferens}\\
		\textit{Ectocarpus siliculosus}\\
		\textit{Schizosaccharomyces pombe}\\
		\textit{Bacillus cereus ATCC 14579}\\
		\textit{Escherichia coli str. K-12 substr. MG1655}\\
		\textit{Escherichia coli O157 H7 str. Sakai}\\
		\textit{Salmonella enterica subsp. enterica serovar Typhi str. CT18}\\
		\textit{Amycolatopsis mediterranei U32}\\
		\textit{Aquifex aeolicus VF5}\\
		\textit{Borrelia burgdorferi B31}\\
		\textit{Chlamydophila pneumoniae CWL029}\\
		\textit{Chlorobium tepidum TLS}\\
		\textit{Deinococcus radiodurans R2}\\
		\textit{Caulobacter crescentus CB15}\\
		\textit{Sulfolobus islandicus M.14.25}\\
		\textit{Nanoarchaeum equitans Kin4-M}\\
		\textit{Haloferax mediterranei ATCC 33500}\\
		\textit{Methanococcus maripaludis S2}\\
		\textit{Cenarchaeum symbiosum A}\\
	\hline
	\end{tabular}
	\caption{Table of genomes using in transcript binning pipeline.
		Genomes were chosen to be a representative of the sampled diversity of the 
		eukaryotic tree of life as possible}
	\label{tab:appen_genomes}
\end{table}

%\subsection{Optimised Classifier Test Data Reports}
%\begin{table}
%\begin{tabularx}{\textwidth}{|X|X|X|X|X|}
%\hline
%\multicolumn{5}{c}{KNeighborsClassifier (algorithm='auto', leaf\_size=30, metric='minkowski', metric\_params=None, n\_neighbors=50, p=2, weights='uniform')} \\
%\hline
%\textbf{class}		& \textbf{precision}  &  \textbf{recall}  & \textbf{f1-score}  &  \textbf{support} \\
%unknown   	&    0.66 	 &   0.49   &    0.56   &   156   \\
%food      	&    0.96 	 &   0.95   &    0.95   &   426   \\
%host      	&    0.92    &   0.96   &    0.94   &   787   \\
%endosymbiont&    0.88    &   0.90   &    0.89   &   359   \\
%\hline
%avg / total &    0.90    &   0.90   &    0.90   &   1728  \\
%\hline
%\end{tabularx}
%\end{table}
%
%\begin{table}
%	\begin{tabular}{|r|c|c|c|c|}
%		\hline
%		\multicolumn{5}{c}{LinearSVC(C=1.0, class\_weight=None, dual=True, fit\_intercept=True,
%			intercept\_scaling=1, loss='squared\_hinge', max\_iter=1000,
%			multi\_class='ovr', penalty='l2', tol=0.0001)}\\
%		\hline
%		\textbf{class}		& \textbf{precision}  &  \textbf{recall}  & \textbf{f1-score}  &  \textbf{support} \\
%		unknown   	&     0.61 	 &    0.23  &     0.33   &   156   \\
%		food      	&    0.96 	 &   0.94   &    0.95   &   426   \\
%		host      	&    0.92    &   0.97   &    0.92   &   787   \\
%		endosymbiont&    0.87    &   0.90   &    0.90   &   359   \\
%		\hline
%		avg / total &   0.87    &   0.88   &    0.88   &   1728  \\
%		\hline
%	\end{tabular}
%\end{table}
%
%
%\begin{table}
%	\begin{tabular}{|r|c|c|c|c|}
%		\hline
%		\multicolumn{5}{c}{SVC(C=1.0, cache\_size=200, class\_weight=None, coef0=0.0, degree=3, gamma=0.3,
%			kernel='rbf', max\_iter=-1, probability=False,
%			shrinking=True, tol=0.001))}\\
%		\hline
%		\textbf{class}		& \textbf{precision}  &  \textbf{recall}  & \textbf{f1-score}  &  \textbf{support} \\
%		unknown   	&    0.51 	 &   0.37   &    0.43   &   156   \\
%		food      	&    0.67 	 &   0.30   &    0.42   &   426   \\
%		host      	&    0.67    &   0.97   &    0.79   &   787   \\
%		endosymbiont&    0.90    &   0.90   &    0.73   &   359   \\
%		\hline
%		avg / total &    0.71    &   0.70   &    0.67   &   1728  \\
%		\hline
%	\end{tabular}
%\end{table}
%
%
%\begin{table}
%	\begin{tabular}{|r|c|c|c|c|}
%		\hline
%		\multicolumn{5}{c}{DecisionTreeClassifier(class\_weight=None, criterion='gini', max\_depth=None,
%			max\_features=None, max\_leaf\_nodes=None, min\_samples\_leaf=1,
%			min\_samples\_split=2, min\_weight\_fraction\_leaf=0.0,
%			random\_state=None, splitter='best')}\\
%		\hline
%		\textbf{class}		& \textbf{precision}  &  \textbf{recall}  & \textbf{f1-score}  &  \textbf{support} \\
%		unknown   	&    0.45 	 &   0.35   &    0.39   &   156   \\
%		food      	&    0.91 	 &   0.93   &    0.92   &   426   \\
%		host      	&    0.88    &   0.92   &    0.90   &   787   \\
%		endosymbiont&    0.82    &   0.81   &    0.81   &   359   \\
%		\hline
%		avg / total &    0.84    &   0.85   &    0.84   &   1728  \\
%		\hline
%	\end{tabular}
%\end{table}
%
%
%
%\begin{table}
%	\begin{tabular}{|r|c|c|c|c|}
%		\hline
%		\multicolumn{5}{c}{ExtraTreeClassifier(class\_weight=None, criterion='gini', max\_depth=None,
%			max\_features='auto', max\_leaf\_nodes=None, min\_samples\_leaf=1,
%			min\_samples\_split=2, min\_weight\_fraction\_leaf=0.0, splitter='random')}\\
%		\hline
%		\textbf{class}		& \textbf{precision}  &  \textbf{recall}  & \textbf{f1-score}  &  \textbf{support} \\
%		unknown   	&    0.43 	 &   0.37   &    0.39   &   156   \\
%		food      	&    0.90 	 &   0.93   &    0.92   &   426   \\
%		host      	&    0.89    &   0.90   &    0.89   &   787   \\
%		endosymbiont&    0.82    &   0.81   &    0.82   &   359   \\
%		\hline
%		avg / total &    0.84    &   0.84   &    0.84   &   1728  \\
%		\hline
%	\end{tabular}
%\end{table}
%
%
%
%\begin{table}
%	\begin{tabular}{|r|c|c|c|c|}
%		\hline
%		\multicolumn{5}{c}{RandomForestClassifier(bootstrap=True, class\_weight=None, criterion='gini', max\_depth=None, max\_features='auto', max\_leaf\_nodes=None, min\_samples\_leaf=1, min\_samples\_split=2, min\_weight\_fraction\_leaf=0.0, n\_estimators=50, n\_jobs=1, oob\_score=False, warm\_start=False)}\\
%		\hline
%		\textbf{class}		& \textbf{precision}  &  \textbf{recall}  & \textbf{f1-score}  &  \textbf{support} \\
%		unknown   	&    0.43 	 &   0.33   &    0.38   &   156   \\
%		food      	&    0.92 	 &   0.95   &    0.94   &   426   \\
%		host      	&    0.89    &   0.93   &    0.91   &   787   \\
%		endosymbiont&    0.84    &   0.81   &    0.82   &   359   \\
%		\hline
%		avg / total &    0.84    &   0.85   &    0.84   &   1728  \\
%		\hline
%	\end{tabular}
%\end{table}
%
%
%
%
%\begin{table}
%	\begin{tabular}{|r|c|c|c|c|}
%		\hline
%		\multicolumn{5}{c}{AdaBoostClassifier(algorithm='SAMME.R', base\_estimator=None, learning\_rate=1.0, n\_estimators=50)}\\
%		\hline
%		\textbf{class}		& \textbf{precision}  &  \textbf{recall}  & \textbf{f1-score}  &  \textbf{support} \\
%		unknown   	&    0.60 	 &   0.39   &    0.47   &   156   \\
%		food      	&    0.96 	 &   0.95   &    0.95   &   426   \\
%		host      	&    0.90    &   0.96   &    0.93   &   787   \\
%		endosymbiont&    0.88    &   0.89   &    0.88   &   359   \\
%		\hline
%		avg / total &    0.88    &   0.89   &    0.88   &   1728  \\
%		\hline
%	\end{tabular}
%\end{table}
%
%
%
%\begin{table}
%	\begin{tabular}{|r|c|c|c|c|}
%		\hline
%		\multicolumn{5}{c}{LDA()}\\
%		\hline
%		\textbf{class}		& \textbf{precision}  &  \textbf{recall}  & \textbf{f1-score}  &  \textbf{support} \\
%		unknown   	&    0.00 	 &   0.39   &    0.00   &   156   \\
%		food      	&    0.97 	 &   0.72  &    0.83   &   426   \\
%		host      	&    0.56    &   1.00   &    0.72   &   787   \\
%		endosymbiont&    0.00    &   0.00   &    0.00   &   359   \\
%		\hline
%		avg / total &    0.49    &   0.63   &    0.53   &   1728  \\
%		\hline
%	\end{tabular}
%\end{table}
%
%\begin{table}
%	\begin{tabular}{|r|c|c|c|c|}
%		\hline
%		\multicolumn{5}{c}{QDA()}\\
%		\hline
%		\textbf{class}		& \textbf{precision}  &  \textbf{recall}  & \textbf{f1-score}  &  \textbf{support} \\
%		unknown   	&    0.10 	 &   0.97   &    0.18   &   156   \\
%		food      	&    0.96 	 &   0.36   &    0.52   &   426   \\
%		host      	&    1.00    &   0.03   &    0.06   &   787   \\
%		endosymbiont&    0.67    &   0.01   &    0.01   &   359   \\
%		\hline
%		avg / total &    0.84    &   0.19   &    0.17   &   1728  \\
%		\hline
%	\end{tabular}
%\end{table}
%
%
%
%\begin{table}
%	\begin{tabular}{|r|c|c|c|c|}
%		\hline
%		\multicolumn{5}{c}{GaussianNB()}\\
%		\hline
%		\textbf{class}		& \textbf{precision}  &  \textbf{recall}  & \textbf{f1-score}  &  \textbf{support} \\
%		unknown   	&    0.10 	 &   0.97   &    0.18   &   156   \\
%		food      	&    0.96 	 &   0.36   &    0.52   &   426   \\
%		host      	&    1.00    &   0.03   &    0.06   &   787   \\
%		endosymbiont&    0.67    &   0.01   &    0.01   &   359   \\
%		\hline
%		avg / total &    0.84    &   0.19   &    0.17   &   1728  \\
%		\hline
%	\end{tabular}
%\end{table}
%
%
%
%\begin{table}
%	\begin{tabular}{|r|c|c|c|c|}
%		\hline
%		\multicolumn{5}{c}{LogisticRegression(C=1.0, class\_weight=None, dual=False, fit\_intercept=True)}\\
%		\hline
%		\textbf{class}		& \textbf{precision}  &  \textbf{recall}  & \textbf{f1-score}  &  \textbf{support} \\
%		unknown   	&    0.00 	 &   0.00   &    0.00   &   156   \\
%		food      	&    0.78 	 &   0.96   &    0.86   &   426   \\
%		host      	&    1.91    &   0.96   &    0.94   &   787   \\
%		endosymbiont&    0.88    &   0.90   &    0.89   &   359   \\
%		\hline
%		avg / total &    0.79    &   0.86   &    0.82   &   1728  \\
%		\hline
%	\end{tabular}
%\end{table}

\section{eDicer}


\begin{table}
    \begin{tabular}{|c|}
        \hline
\textbf{eDicer Eukaryote Predicted Transcriptomes}\\
\hline
\textit{Acanthamoeba castellanii} str. Neff\\
\textit{Aplanochytrium kerguelense} PBS07\\
\textit{Arabidopsis thaliana}\\
\textit{Aurantiochytrium limacinum}\\
\textit{Batrachochytrium dendrobatidis} JAM81\\
\textit{Bigelowiella natans}\\
\textit{Blastocystis hominis}\\
\textit{Bodo saltans}\\
\textit{Caenorhabditis elegans}\\
\textit{Capsaspora owczarzaki}\\
\textit{Chlamydomonas reinhardtii}\\
\textit{Chondrus crispus}\\
\textit{Ciona intestinalis}\\
\textit{Cryptococcus neoformans} var. grubii H99\\
\textit{Cryptosporidium parvum}\\
\textit{Cyanidioschyzon merolae}\\
\textit{Cyanophora paradoxa}\\
\textit{Dictyostelium discoideum}\\
\textit{Drosophila melanogaster}\\
\textit{Ectocarpus siliculosus}\\
\textit{Emiliania huxleyi} CCMP1516\\
\textit{Entamoeba histolytica}\\
\textit{Fonticula alba}\\
\textit{Giardia intestinalis}\\
\textit{Guillardia theta}\\
\textit{Homo sapiens}\\
\textit{Hyaloperonospora arabidopsidis}\\
\textit{Klebsormidium flaccidum}\\
\textit{Laccaria bicolor}\\
\textit{Monosiga brevicollis}\\
\textit{Mortierella verticillata} NRRL 6337\\
\textit{Mus musculus}\\
\textit{Naegleria gruberi}\\
\textit{Nannochloropsis gaditana}\\
\textit{Neurospora crassa} OR74A\\
\textit{Ostreococcus lucimarinus}\\
\textit{Perkinsus marinus}\\
\textit{Phaeodactylum tricornutum}\\
\textit{Physcomitrella patens}\\
\textit{Phytophthora ramorum}\\
\textit{Plasmodium falciparum}\\
\textit{Populus trichocarpa}\\
\textit{Reticulomyxa filosa}\\
\textit{Rozella allomycis} CSF55\\
\textit{Salpingoeca} sp. ATCC 50818\\
\textit{Schizosaccharomyces pombe}\\
\textit{Sphaeroforma arctica} jp610\\
\textit{Takifugu rubripes}\\
\textit{Tetrahymena thermophila macronucleus}\\
\textit{Thalassiosira pseudonana}\\
\textit{Thecamonas trahens} ATCC 50062\\
\textit{Toxoplasma gondii}\\
\textit{Trichomonas vaginalis} G3\\
\textit{Trichoplax adhaerens}\\
\textit{Trypanosoma brucei}\\
\textit{Tuber melanosporum}\\
\textit{Ustilago maydis}\\
\textit{Vitrella brassicaformis} CCMP3155\\
\hline
\end{tabular}
\caption[]{}
\label{tab:edicer_genomes_euks}
\end{table}

\begin{table}
    \begin{tabular}{|c|}
        \hline
        \textbf{eDicer Bacteria Predicted Transcriptomes} \\
        \hline
        \textit{Acidimicrobium ferrooxidans} DSM 10331\\
\textit{Nocardia farcinica} IFM 10152\\
\textit{Frankia alni} ACN14a \\
\textit{Propionibacterium acidifaciens} DSM 21887\\
\textit{Kitasatospora setae} KM-6054\\
\textit{Bifidobacterium longum} NCC2705\\
\textit{Collinsella tanakaei} YIT 12063 \\
\textit{Rubrobacter xylanophilus} DSM 9941\\
\textit{Conexibacter woesei} DSM 14684\\
\textit{Moorella thermoacetica} ATCC 39073\\
\textit{Aquifex aeolicus} VF5\\
\textit{Persephonella marina} EX-H1\\
\textit{Desulfurobacterium thermolithotrophum} DSM 11699\\
\textit{Hydrogenobacter thermophilus} TK-6\\
\textit{Chthonomonas calidirosea} T49\\
\textit{Fimbriimonas ginsengisoli} Gsoil 348\\
\textit{Armatimonadetes bacterium} CSP1-3\\
\textit{Bacteroides fragilis} 3\_1\_12\\
\textit{Flavobacterium psychrophilum} JIP02\_86\\
\textit{Salinibacter ruber} DSM 13855\\
\textit{Chlorobium tepidum} TLS\\
\textit{Chlorobium luteolum} DSM 273\\
\textit{Chloroherpeton thalassium} ATCC 35110\\
\textit{Prosthecochloris aestuarii} DSM 271\\
\textit{Ignavibacterium album} JCM 16511\\
\textit{Melioribacter roseus} P3M-2\\
\textit{Phycisphaera mikurensis} NBRC 102666\\
\textit{Isosphaera pallida} ATCC 43644\\
\textit{Blastopirellula marina} DSM 3645\\
\textit{Chlamydia trachomatis} 434/Bu\\
\textit{Waddlia chondrophila} WSU 86-1044\\
\textit{Lentisphaera araneosa} HTCC2155\\
\textit{Opitutus terrae} PB90-1\\
\textit{Akkermansia muciniphila} ATCC BAA-835\\
\textit{Anaerolinea thermophila} UNI-1\\
\textit{Caldilinea aerophila} DSM 14535\\
\textit{Chloroflexus aurantiacus} J-10-fl\\
\textit{Dehalococcoides mccartyi} 195\\
\textit{Ktedonobacter racemifer} DSM 44963\\
\textit{Thermomicrobium roseum} DSM 5159\\
\textit{Gloeobacter violaceus} PCC 7421\\
\textit{Nostoc punctiforme} PCC 73102\\
\textit{Acaryochloris} sp. CCMEE 5410\\
\textit{Synechococcus elongatus} PCC 6301\\
\textit{Synechococcus} sp. JA-3-3Ab\\
\textit{Chroococcidiopsis thermalis} PCC 7203\\
\textit{Pleurocapsa} sp. PCC 7327\\
\textit{Fischerella muscicola} PCC 7414\\
\textit{Calditerrivibrio nitroreducens} DSM 19672\\
\textit{Deferribacter desulfuricans} SSM1\\
\textit{Denitrovibrio acetiphilus} DSM 12809\\
\textit{Flexistipes sinusarabici} DSM 4947\\
\textit{Deinococcus radiodurans} R1\\
\textit{Truepera radiovictrix} DSM 17093\\
\textit{Meiothermus silvanus} DSM 9946\\
\textit{Thermus aquaticus} Y51MC23\\
\textit{Acidobacterium capsulatum} ATCC 51196\\
\textit{Fibrobacter succinogenes} subsp. succinogenes S85\\
\textit{Chitinivibrio alkaliphilus} ACht1\\
\textit{Bacillus subtilis} B7-s\\
\textit{Lactobacillus acidophilus} NCFM\\
\textit{Streptococcus suis} BM407\\
\textit{Halanaerobium praevalens} DSM 2228\\
\textit{Coprobacillus} sp. 3\_3\_56FAA\\
\textit{Mitsuokella multacida} DSM 20544\\
\textit{Anaerococcus hydrogenalis} ACS-025-V-Sch4\\
\textit{Thermoanaerobacter ethanolicus} JW 200\\
\textit{Cetobacterium} sp. ZOR0034\\
\textit{Fusobacterium nucleatum} 13\_3C\\
\textit{Ilyobacter polytropus} DSM 2926\\
\textit{Leptotrichia wadei} F0279\\
\textit{Sebaldella termitidis} ATCC 33386\\
\textit{Nitrospira defluvii}\\
\textit{Thermodesulfovibrio islandicus} DSM 12570\\
\textit{Leptospirillum} sp. Group I\\
\textit{Caulobacter crescentus} CB15\\
\textit{Agrobacterium tumefaciens} WRT31\\
\textit{Hyphomonas} sp. 25B14\_1\\
\textit{Parvularcula bermudensis} HTCC2503\\
\textit{Wolbachia pipientis} wAlbB\\
\textit{Sphingobium} sp. AP49\\
\textit{Acetobacter pasteurianus} 386B\\
\textit{Bordetella avium} 197N\\
\textit{Neisseria bacilliformis} ATCC BAA-1200\\
\textit{Azoarcus} sp. BH72\\
\textit{Nitrosomonas eutropha} C91\\
\textit{Methylobacillus flagellatus} KT\\
\textit{Aeromonas hydrophila} NJ-35\\
\textit{Escherichia coli} str. K-12\\
\textit{Pseudomonas aeruginosa} PAO1\\
\textit{Xanthomonas oryzae} ATCC 35933\\
\textit{Alcanivorax} sp. DG881\\
\textit{Vibrio fischeri} SR5\\
\textit{Shewanella baltica} BA175\\
\textit{Bdellovibrio bacteriovorus} HD100\\
\textit{Desulfarculus baarsii} DSM 2075\\
\textit{Myxococcus xanthus} DZ2\\
\textit{Desulfovibrio vulgaris} DP4\\
\textit{Geobacter sulfurreducens} KN400\\
\textit{Campylobacter jejuni} 10186\\
\textit{Helicobacter pylori} 35A\\
\textit{Nautilia profundicola} AmH\\
\textit{Nitratifractor salsuginis} DSM 16511\\
\textit{Sulfurovum lithotrophicum}\\
\textit{Brachyspira innocens} ATCC 29796\\
\textit{Leptospira interrogans} serovar Bim str. P2529\\
\textit{Borrelia burgdorferi} B31\\
\textit{Spirochaeta thermophila} DSM 6578\\
\textit{Treponema denticola} ATCC 33520\\
\textit{Synergistes} sp. 3\_1\_syn1\\
\textit{Aminomonas paucivorans} DSM 12260\\
\textit{Dethiosulfovibrio peptidovorans} DSM 11002\\
\textit{Anaerobaculum mobile} DSM 13181\\
\textit{Pyramidobacter piscolens} W5455\\
\textit{Acholeplasma laidlawii} PG-8A\\
\textit{Mycoplasma mycoides} subsp. mycoides SC str. PG1\\
\textit{Kosmotoga olearia} TBF 19.5.1\\
\textit{Petrotoga mobilis} SJ95\\
\textit{Fervidobacterium islandicum}\\
\textit{Caldisericum exile} AZM16c01\\
\textit{Desulfurispirillum indicum} S5\\
\textit{Dictyoglomus thermophilum} H-6-12\\
\textit{Elusimicrobium minutum} Pei191\\
\textit{Gemmatimonas aurantiaca} T-27\\
\textit{Gemmatimonadetes bacterium} KBS708\\
\textit{Nitrospina gracilis} 3/211\\
\textit{Mariprofundus ferrooxydans} PV-1\\
\textit{Thermodesulfatator atlanticus} DSM 21156\\
\textit{Thermodesulfobacterium thermophilum} DSM 1276\\
\textit{Caldithrix abyssi} DSM 13497\\
\hline
\end{tabular}
\caption[]{}
\label{tab:edicer_genome_bacteria}
\end{table}


\begin{table}
    \begin{tabular}{|c|}
        \hline
        \textbf{eDicer Archaea Predicted Transcriptomes}\\
        \hline
\textit{Acidianus hospitalis} W1\\
\textit{Acidilobus saccharovorans} 345-15\\
\textit{Aciduliprofundum boonei} T469\\
\textit{Aeropyrum pernix} K1\\
\textit{Archaeoglobus fulgidus} DSM 4304\\
\textit{Caldisphaera lagunensis} DSM 15908\\
\textit{Caldivirga maquilingensis} IC-167\\
\textit{Candidatus Korarchaeum cryptofilum} OPF8\\
\textit{Candidatus Nitrosopelagicus brevis}\\
\textit{Cenarchaeum symbiosum A}\\
\textit{Desulfurococcus fermentans} DSM 16532\\
\textit{Ferroglobus placidus} DSM 10642\\
\textit{Ferroplasma acidarmanus} fer1\\
\textit{Fervidicoccus fontis} Kam940\\
\textit{Halalkalicoccus jeotgali} B3\\
\textit{Haloarcula hispanica} ATCC 33960\\
\textit{Halobacterium salinarum} R1\\
\textit{Haloferax mediterranei} ATCC 33500\\
\textit{Halogeometricum borinquense} DSM 11551\\
\textit{Halomicrobium mukohataei} DSM 12286\\
\textit{Halopiger xanaduensis} SH-6\\
\textit{Haloquadratum walsbyi} DSM 16790\\
\textit{Halorhabdus utahensis} DSM 12940\\
\textit{Halorubrum lacusprofundi} ATCC 49239\\
\textit{Haloterrigena turkmenica} DSM 5511\\
\textit{Halovivax ruber} XH-70\\
\textit{Hyperthermus butylicus} DSM 5456\\
\textit{Ignicoccus hospitalis} KIN4 I\\
\textit{Ignisphaera aggregans} DSM 17230\\
\textit{Metallosphaera sedula} DSM 5348\\
\textit{Methanobacterium formicicum} DSM 3637\\
\textit{Methanobrevibacter ruminantium} M1\\
\textit{Methanobrevibacter} sp. AbM4\\
\textit{Methanocaldococcus fervens} AG86\\
\textit{Methanocella conradii} HZ254\\
\textit{Methanococcoides burtonii} DSM 6242\\
\textit{Methanococcus maripaludis} S2\\
\textit{Methanocorpusculum labreanum} Z\\
\textit{Methanoculleus bourgensis} MS2\\
\textit{Methanofollis liminatans} DSM 4140\\
\textit{Methanohalobium evestigatum} Z-7303\\
\textit{Methanohalophilus mahii} DSM 5219\\
\textit{Methanolobus psychrophilus} R15\\
\textit{Methanomethylovorans hollandica} DSM 15978\\
\textit{Methanoplanus limicola} DSM 2279\\
\textit{Methanopyrus kandleri} AV19\\
\textit{Methanoregula formicicum} SMSP\\
\textit{Methanosaeta thermophila} PT\\
\textit{Methanosalsum zhilinae} DSM 4017\\
\textit{Methanosarcina acetivorans} C2A\\
\textit{Methanosphaera stadtmanae} DSM 3091\\
\textit{Methanosphaerula palustris} E1-9c\\
\textit{Methanospirillum hungatei} JF-1\\
\textit{Methanothermobacter marburgensis} str. Marburg\\
\textit{Methanothermococcus okinawensis} IH1\\
\textit{Methanothermus fervidus} DSM 2088\\
\textit{Methanotorris igneus} Kol 5\\
\textit{Nanoarchaeum equitans} Kin4-M\\
\textit{Natrialba magadii} ATCC 43099\\
\textit{Natrinema pellirubrum} DSM 15624\\
\textit{Natronobacterium gregoryi} SP2\\
\textit{Natronococcus occultus} SP4\\
\textit{Natronomonas pharaonis} DSM 2160\\
\textit{Nitrosopumilus maritimus} SCM1\\
\textit{Picrophilus torridus} DSM 9790\\
\textit{Pyrobaculum arsenaticum} DSM 13514\\
\textit{Pyrococcus yayanosii} CH1\\
\textit{Pyrolobus fumarii} 1A\\
\textit{Salinarchaeum} sp. Harcht-Bsk1\\
\textit{Staphylothermus hellenicus} DSM 12710\\
\textit{Sulfolobus islandicus} M.14.25\\
\textit{Thermococcus barophilus} MP\\
\textit{Thermofilum pendens} Hrk 5\\
\textit{Thermogladius cellulolyticus} 1633\\
\textit{Thermoplasma acidophilum} DSM 1728\\
\textit{Thermoplasmatales archaeon} BRNA1\\
\textit{Thermoproteus uzoniensis} 768-20\\
\textit{Thermosphaera aggregans} DSM 11486\\
\textit{Vulcanisaeta distributa} DSM 14429\\
\hline
\end{tabular}
\caption[]{}
\label{tab:edicer_archaea}
\end{table}



