\nomenclature{\(\theta\)}{Parameters}
\nomenclature{\(\mathcal{D}\)}{Training Data}
\nomenclature{\(\mathbb{R}\)}{Real Numbers}
\nomenclature{\(\mathbb{Z}\)}{Integers}
\nomenclature{AA}{Amino Acid}
\nomenclature{eToL}{Eukaryotic Tree of Life}
\nomenclature{ToL}{Tree of Life}
\nomenclature{AdaBoost}{Adaptive Boosting}
\nomenclature{ABC}{ATP-Binding Cassette}
\nomenclature{AIC}{Akaike Information Criterion}
\nomenclature{ALE}{Assembly Likelihood Evaluation}
\nomenclature{AGO}{Argonaute proteins}
\nomenclature{AMP}{Adenosine Monophosphate}
\nomenclature{ATCC}{American Type Culture Collection}
\nomenclature{ATP}{Adenosine Triphosphate}
\nomenclature{BIC}{Bayesian Information Criterion}
\nomenclature{BLAST}{Basic Local Alignment Search Tool}
\nomenclature{BLASTN}{Nucleotide BLAST}
\nomenclature{BLASTP}{Protein BLAST}
\nomenclature{BLASTX}{BLAST against a Protein Database using a Translated Nucleotide Query}
\nomenclature{BLAT}{BLAST-Like Alignment Tool}
\nomenclature{bp}{Base Pair}
\nomenclature{BLOSUM}{Blocks Substitution Matrix}
\nomenclature{CCAP}{Culture Collection of Algae and Protozoa}
\nomenclature{CDF}{Cumulative Distribution Function}
\nomenclature{CD-HIT}{Clustering Database at High Identity with Tolerance}
\nomenclature{CDS}{Coding DNA Sequence}
\nomenclature{CGAT}{Computational Genomics Analysis and Training}
\nomenclature{CI}{Continuous Integration}
\nomenclature{CNN}{Convolutional Neural Network}
\nomenclature{CONCOCT}{Clustering Contigs with Coverage and Composition}
\nomenclature{CPU}{Central Processing Unit}
\nomenclature{CRF}{Conditional Random Fields}
\nomenclature{CRISPR}{Clustered Regularly Interspaced Short Palindromic Repeats}
\nomenclature{CTAB}{Cetrimonium bromide}
\nomenclature{CV}{Cross-Validation}
\nomenclature{DAPI}{4',6-diamidino-2-phenylindole}
\nomenclature{dBG}{de Bruijn graph}
\nomenclature{DBSCAN}{Density-Based Spatial Clustering of Applications with Noise}
\nomenclature{DCMU}{3-(3,4-dichlorophenyl)-1,1-dimethylurea}
\nomenclature{DDBJ}{DNA Data Bank of Japan}
\nomenclature{ddNTP}{Dideoxy Nucleoside Triphosphate}
\nomenclature{dNTP}{Deoxy Nucleoside Triphosphate}
\nomenclature{DETONATE}{\textit{de novo} Transcriptome RNA-Seq Assembly with or without Truth Evaluation}
\nomenclature{DNA}{Deoxyribonucleic Acid}
\nomenclature{DPP}{Dirichlet Process Prior}
\nomenclature{DT}{Decision trees}
\nomenclature{DV}{Digestive Vacuole}
\nomenclature{EGT}{Endosymbiotic Gene Transfer}
\nomenclature{EIC}{Extracted-Ion Chromatogram}
\nomenclature{EM}{Expectation-Maximisation}
\nomenclature{EPP}{Eulerian Path Problem}
\nomenclature{ER}{Endoplasmic Reticulum}
\nomenclature{EST}{Expressed Sequence Tags}
\nomenclature{ETE}{Environment for Tree Exploration}
\nomenclature{ETS}{External Transcribed Spacers}
\nomenclature{FAT}{Fatty Acid Translocation}
\nomenclature{FDR}{False Discovery Rate}
\nomenclature{FFT}{Fast Fourier Transform}
\nomenclature{FN}{False Negative}
\nomenclature{FP}{False Positive}
\nomenclature{FRET}{Fluorescence Resonance Energy Transfer}
\nomenclature{GAII}{Genome Analyzer 2}
\nomenclature{GC}{Guanine Cysteine}
\nomenclature{GC-}{Gas Chromatography}
\nomenclature{GFF}{General Feature Format}
\nomenclature{GFP}{Green Fluorescent Protein}
\nomenclature{GI}{Gene Identifier}
\nomenclature{GMM}{Gaussian Mixture Model}
\nomenclature{GNU}{GNU's not UNIX}
\nomenclature{GO}{Gene Ontology}
\nomenclature{GTF}{General Transfer Format}
\nomenclature{GTR}{Generalised Time Reversible model}
\nomenclature{HAMAP}{High-quality Automated and Manual Annotation of Proteins}
\nomenclature{HGT}{Horizontal Gene Transfer}
\nomenclature{HMM}{Hidden Markov Model}
\nomenclature{HPP}{Hamiltonian Path Problem}
\nomenclature{HPLC}{High Performance Liquid Chromatography}
\nomenclature{KAAS}{KEGG Automatic Annotation Server}
\nomenclature{IDBA}{Iterative de Bruijn Assembler}
\nomenclature{IDBA-UD}{Iterative de Bruijn Assembler - Uneven Sequencing Depth}
\nomenclature{IDBA-MT}{Iterative de Bruijn Assembler - Metatranscriptome}
\nomenclature{IDBA-MTP}{Iterative de Bruijn Assembler - Metatranscriptome with Protein Alignment}
\nomenclature{IES}{Internal Eliminated Sequences}
\nomenclature{iid}{Independent and Identically Distributed}
\nomenclature{IPTG}{Isopropyl \(\beta\)-D-1-thiogalactopyranoside}
\nomenclature{IR}{Infrared}
\nomenclature{ITS}{Internal Transcribed Spacer}
\nomenclature{JGI}{Joint Genome Institute}
\nomenclature{KAT}{K-mer Analysis Toolkit}
\nomenclature{KDE}{Kernel Density Estimate}
\nomenclature{KEGG}{Kyoto Encyclopedia of Genes and Genomes}
\nomenclature{LB}{Luria Broth}
\nomenclature{LBA}{Long Branch Attraction}
\nomenclature{LC-}{Liquid Chromatography}
\nomenclature{LCA}{Lowest Common Ancestor}
\nomenclature{LDA}{Linear Discriminant Analysis}
\nomenclature{LECA}{Last Eukaryote Common Ancestor}
\nomenclature{LG}{Le-Gascuel model}
\nomenclature{LTS}{Long-Term Support}
\nomenclature{MAC}{Macronucleus}
\nomenclature{MAFFT}{Multiple Alignment using Fast Fourier Transform}
\nomenclature{MCMC}{Markov Chain Monte Carlo}
\nomenclature{MCMCMC}{Metropolis-Coupled MCMC}
\nomenclature{MDA}{Multiple Displacement Amplification}
\nomenclature{MDS}{Multidimensional Scaling}
\nomenclature{MEGAN}{Metagenome Analyzer}
\nomenclature{METLIN}{Metabolite Link Database}
\nomenclature{MFS}{Major Facilitator Superfamily}
\nomenclature{MIC}{Micronucleus}
\nomenclature{ML}{Maximum Likelihood}
\nomenclature{MLE}{Maximum Likelihood Estimator}
\nomenclature{MNIST}{Mixed National Institute of Standards and Technology}
\nomenclature{MP}{Maximum Parsimony}
\nomenclature{MPP}{Mitochondria and Plastid Porin family}
\nomenclature{MS}{Mass Spectrometry}
\nomenclature{MSA}{Multiple Sequence Alignment}
\nomenclature{MUSCLE}{Multiple Sequence Comparison by Log-Expectation}
\nomenclature{mRNA}{Messenger RNA}
\nomenclature{siRNA}{Short-Interfering RNA}
\nomenclature{miRNA}{Micro RNA}
\nomenclature{ncRNA}{Non-Coding RNA}
\nomenclature{piRNA}{Piwi-Interacting RNA}
\nomenclature{scnRNA}{Scan RNA}
\nomenclature{ssRNA}{Single-Stranded RNA}
\nomenclature{dsRNA}{Double-Stranded RNA}
\nomenclature{NADH}{Nicotinamide adenine dinucleotide (reduced)}
\nomenclature{NBRP}{National BioResource Project}
\nomenclature{NCBI}{National Center for Biotechnology Information}
\nomenclature{NCL}{New Cereal Leaf-Prescott}
\nomenclature{ND}{NanoDrop}
\nomenclature{NIST}{National Institute of Standards and Technology}
\nomenclature{NJ}{Neighbour-Joining}
\nomenclature{NP}{Non-deterministic Polynomial Time}
\nomenclature{NMR}{Nuclear Magnetic Resonance}
\nomenclature{NNI}{Nearest Neighbour Interchange}
\nomenclature{NR}{Non-Redundant}
\nomenclature{NTS}{Non-transcribed Spacers}
\nomenclature{OLC}{Overlap Consensus}
\nomenclature{ORF}{Open Reading Frame}
\nomenclature{P}{Polynomial Time}
\nomenclature{PAM}{Protein Data Bank}
\nomenclature{PBCV}{\textit{Paramecium bursaria}-\textit{Chlorella} Virus}
\nomenclature{PbMr}{\textit{Paramecium bursaria}-\textit{Micractinium reisseri}}
\nomenclature{PCA}{Principal Component Analysis}
\nomenclature{PCR}{Polymerase Chain Reaction}
\nomenclature{PDB}{Protein Data Bank}
\nomenclature{PE}{Paired-End}
\nomenclature{PFAM}{Protein Families Database}
\nomenclature{PGM}{PiggyMac}
\nomenclature{PP}{Posterior Probability}
\nomenclature{PTGS}{Post-Transcription Gene Silencing}
\nomenclature{PV}{Perialgal Vacuole}
\nomenclature{QC}{Quality Control}
\nomenclature{qPCR}{Quantitative Polymerase Chain Reaction}
\nomenclature{QDA}{Quadratic Discriminant Analysis}
\nomenclature{QQQ}{Triple Quadrupole}
\nomenclature{QTOF}{Quadrupole Time-of-Flight}
\nomenclature{QUAST}{Quality Assessment Tool for Genome Assemblies}
\nomenclature{RAM}{Random Access Memory}
\nomenclature{RBF}{Radial Basis Function}
\nomenclature{RBM}{Restricted Boltzmann Machine}
\nomenclature{RdRP}{RNA-Dependent RNA polymerase}
\nomenclature{RFO}{Raffinose Family Oligosaccharide}
\nomenclature{RNA}{Ribonucleic Acid}
\nomenclature{RNAi}{RNA-mediated Interference}
\nomenclature{RNase}{Ribonuclease}
\nomenclature{RNN}{Recurrent Neural Network}
\nomenclature{RSEM-EVAL}{RNA-Seq by Expectation-Maximisation - Evaluation}
\nomenclature{RT}{Reverse Transcription}
\nomenclature{RT-PCR}{Reverse Transcription PCR}
\nomenclature{RVM}{Relevance Vector Machine}
\nomenclature{SC}{Single Cell}
\nomenclature{SDC}{Single Digested \textit{Chlorella}}
\nomenclature{SCT}{Single-Cell Transcriptome}
\nomenclature{SEECER}{Sequencing Error Correction for RNA Reads}
\nomenclature{SGC}{Single Green \textit{Chlorella}}
\nomenclature{SGLT}{Sodium-Glucose symporters}
\nomenclature{SI}{International System of Units}
\nomenclature{SMART}{Single Molecular Analysis Research Tool}
\nomenclature{SMRT}{Single Molecule Real-Time}
\nomenclature{SNP}{Single Nucleotide Polymorphism}
\nomenclature{SPR}{Subtree-Prune-and-Regraph}
\nomenclature{SRA}{Sequence Read Archive}
\nomenclature{SSF}{Sodium-Solute Symporter Family}
\nomenclature{SVM}{Support Vector Machine}
\nomenclature{TA}{Thymine-Adenine}
\nomenclature{TCDB}{Transporter Classification Database}
\nomenclature{TM}{Transmembrane}
\nomenclature{TMHMM}{Transmembrane Hidden Markov Model}
\nomenclature{TN}{True Negative}
\nomenclature{TP}{True Positive}
\nomenclature{TPM}{Transcripts Per Million}
\nomenclature{t-SNE}{t-Distributed Stochastic Neighbour Embedding}
\nomenclature{TST}{Tonoplast Sugar Transporter}
\nomenclature{UK}{United Kingdom}
\nomenclature{UTR}{Untranslated Region}
\nomenclature{UV}{Ultraviolet}
\nomenclature{VCS}{Version Control System}
\nomenclature{WGD}{Whole Genome Duplication}
\nomenclature{WTA}{Whole Transcriptome Amplification}
\nomenclature{XCMS}{X-Chromatography Mass Spectrometry Library}
\nomenclature{\(OD_{600}\)}{Optical Density at 600 nanometre wavelength} 


\printnomenclature[3cm]
