The photosynthetic endosymbioses between \textit{Paramecium bursaria} and its green algal endosymbionts 
(\textit{Chlorella variabilis}, \textit{Chlorella vulgaris}, \textit{Micractinium reisseri} and \textit{Coccomyxa} sp.) 
have long been suggested to represent nascent endosymbiotic interactions. This is because
host and endosymbiont are believed to be able to exist and reproduce separately. 
Understanding the molecular systems underpinning these relationships would therefore 
provide a model system to investigate the process of photosynthetic endosymbioses 
before molecular co-dependence has become fixed (leading to genomic integration).
To this end, the metatranscriptome of \textit{P. bursaria-M. reisseri} during 
lit and dark conditions was recovered using single cell methods. 
This necessitated the development of novel techniques to optimise the 
assembly and the post-assembly attribution of transcripts to their originating organism. 
This work represents the first \textit{de novo} single cell transcriptomic analysis of a multi-member eukaryotic system.  


Metabolic function was then analysed in this system using both transcriptomic and metabolomics data.
This identified potential roles for novel sugar, amino acid, and 
fatty acid interactions in the \textit{M. reisseri} endosymbiosis.
Additionally, \textit{P. bursaria} SW1 was determined to potentially form a obligate host of
\textit{M. reisseri} SW1-ZK.

 
Finally, this work also revealed a putative non-functional exogenous RNA induced 
RNAi system in \textit{P. bursaria} potentially related to the absence of a factor 
associated with uptake of RNA from host vacuoles in both \textit{P. bursaria}
transcriptomes and a partial \textit{P. bursaria} single cell genome.  
An analysis of the level of potential RNAi ``cross-talk'' 
collisions with the active host transcriptome suggests that the function of an 
exogenous RNA induced RNAi system in the presence of a eukaryotic endosymbiont may be deleterious.

 
Therefore, despite discovering several barriers to the utility of these systems for studying
endosymbiotic evolution there is still benefit to their study. The ``omic'' resources and analyses 
presented here offer an important dataset to inform further analysis.
